--- chrony-2.3.ebuild.orig	2020-11-28 21:52:29.000000000 -0800
+++ chrony-2.3.ebuild	2020-11-21 17:03:10.156948446 -0800
@@ -3,7 +3,7 @@
 # $Id$
 
 EAPI=5
-inherit eutils systemd toolchain-funcs
+inherit eutils systemd toolchain-funcs flag-o-matic
 
 DESCRIPTION="NTP client and server programs"
 HOMEPAGE="http://chrony.tuxfamily.org/"
@@ -11,7 +11,7 @@ SRC_URI="http://download.tuxfamily.org/$
 LICENSE="GPL-2"
 SLOT="0"
 
-KEYWORDS="~amd64 ~arm ~hppa ~mips ~ppc ~ppc64 ~sparc ~x86"
+KEYWORDS="~alpha amd64 ~arm hppa ~mips ~ppc ppc64 ~sparc ~x86"
 IUSE="caps +cmdmon ipv6 libedit +ntp +phc +pps readline +refclock +rtc selinux +adns"
 REQUIRED_USE="
 	?? ( libedit readline )
@@ -45,6 +45,11 @@ src_prepare() {
 src_configure() {
 	tc-export CC
 
+	strip-flags
+	filter-flags -O2 -O3 -O4 -Os -O1
+	append-cflags -O0
+	append-cxxflags -O0
+
 	local CHRONY_EDITLINE
 	# ./configure legend:
 	# --disable-readline : disable line editing entirely
