diff -ur sox-14.4.2.orig/src/formats.c sox-14.4.2/src/formats.c
--- sox-14.4.2.orig/src/formats.c	2014-10-27 02:55:50.000000000 +0000
+++ sox-14.4.2/src/formats.c	2016-06-20 06:56:23.588144645 +0000
@@ -422,7 +422,7 @@
   /* To fix this #error, either simply remove the #error line and live without
    * file-type detection with pipes, or add support for your compiler in the
    * lines above.  Test with cat monkey.wav | ./sox --info - */
-  #error FIX NEEDED HERE
+  //#error FIX NEEDED HERE
   #define NO_REWIND_PIPE
   (void)fp;
 #endif
