diff -ur fontconfig-2.11.1.orig/Makefile.in fontconfig-2.11.1/Makefile.in
--- fontconfig-2.11.1.orig/Makefile.in	2016-06-19 01:23:15.930473115 +0000
+++ fontconfig-2.11.1/Makefile.in	2016-06-19 01:24:52.679470017 +0000
@@ -457,20 +457,7 @@
 .SUFFIXES:
 am--refresh: Makefile
 	@:
-$(srcdir)/Makefile.in:  $(srcdir)/Makefile.am  $(am__configure_deps)
-	@for dep in $?; do \
-	  case '$(am__configure_deps)' in \
-	    *$$dep*) \
-	      echo ' cd $(srcdir) && $(AUTOMAKE) --gnu'; \
-	      $(am__cd) $(srcdir) && $(AUTOMAKE) --gnu \
-		&& exit 0; \
-	      exit 1;; \
-	  esac; \
-	done; \
-	echo ' cd $(top_srcdir) && $(AUTOMAKE) --gnu Makefile'; \
-	$(am__cd) $(top_srcdir) && \
-	  $(AUTOMAKE) --gnu Makefile
-Makefile: $(srcdir)/Makefile.in $(top_builddir)/config.status
+Makefile: $(top_builddir)/config.status
 	@case '$?' in \
 	  *config.status*) \
 	    echo ' $(SHELL) ./config.status'; \
diff -ur fontconfig-2.11.1.orig/doc/Makefile.in fontconfig-2.11.1/doc/Makefile.in
--- fontconfig-2.11.1.orig/doc/Makefile.in	2016-06-19 01:23:16.313473103 +0000
+++ fontconfig-2.11.1/doc/Makefile.in	2016-06-19 01:26:03.948467735 +0000
@@ -714,19 +714,7 @@
 
 .SUFFIXES:
 .SUFFIXES: .fncs .sgml .txt .html .c .lo .log .o .obj .pdf .test .test$(EXEEXT) .trs
-$(srcdir)/Makefile.in:  $(srcdir)/Makefile.am  $(am__configure_deps)
-	@for dep in $?; do \
-	  case '$(am__configure_deps)' in \
-	    *$$dep*) \
-	      ( cd $(top_builddir) && $(MAKE) $(AM_MAKEFLAGS) am--refresh ) \
-	        && { if test -f $@; then exit 0; else break; fi; }; \
-	      exit 1;; \
-	  esac; \
-	done; \
-	echo ' cd $(top_srcdir) && $(AUTOMAKE) --gnu doc/Makefile'; \
-	$(am__cd) $(top_srcdir) && \
-	  $(AUTOMAKE) --gnu doc/Makefile
-Makefile: $(srcdir)/Makefile.in $(top_builddir)/config.status
+Makefile: $(top_builddir)/config.status
 	@case '$?' in \
 	  *config.status*) \
 	    cd $(top_builddir) && $(MAKE) $(AM_MAKEFLAGS) am--refresh;; \
@@ -897,8 +885,16 @@
 	  $(MKDIR_P) "$(DESTDIR)$(docdir)" || exit 1; \
 	fi; \
 	for p in $$list; do \
-	  if test -f "$(CURDIR)/$$p"; then d="$(CURDIR)/"; else d="$(srcdir)/"; fi; \
-	  echo "$$d$$p"; \
+	  if [[ -f "$(CURDIR)/$$p" ]]; then \
+		d="$(CURDIR)/"; \
+		echo "$$d$$p"; \
+	  elif [[ -f "$(srcdir)/$$p" ]]; then \
+		d="$(srcdir)/"; \
+		echo "$$d$$p"; \
+	  else \
+		d=""; \
+		echo "$$p"; \
+	  fi; \
 	done | $(am__base_list) | \
 	while read files; do \
 	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(docdir)'"; \
