diff -ur wine-1.8.2.orig/loader/preloader.c wine-1.8.2/loader/preloader.c
--- wine-1.8.2.orig/loader/preloader.c	2016-08-30 03:41:21.255760976 -0700
+++ wine-1.8.2/loader/preloader.c	2016-08-30 03:56:56.855745999 -0700
@@ -369,8 +369,7 @@
                   "movq $thread_data,%rsi\n\t"
                   "movq $0x1002,%rdi\n\t"    /* ARCH_SET_FS */
 #if defined(__X32__)
-                  "movq $0x40000000,%rax\n\t"       /* __X32_SYSCALL_BIT */
-                  "addq $158,%rax\n\t"       /* SYS_arch_prctl */
+                  "movq $0x4000009E,%rax\n\t"       /* SYS_arch_prctl */
 #else
                   "movq $158,%rax\n\t"       /* SYS_arch_prctl */
 #endif
@@ -409,40 +408,40 @@
 
 #if defined(__X32__)
 void wld_exit( int code ) __attribute__((noreturn));
-SYSCALL_NOERR( wld_exit, 0x40000000 + 60 /* SYS_exit */ );
+SYSCALL_NOERR( wld_exit, 0x4000003C /* SYS_exit */ );
 
 ssize_t wld_read( int fd, void *buffer, size_t len );
-SYSCALL_FUNC( wld_read, 0x40000000 + 0 /* SYS_read */ );
+SYSCALL_FUNC( wld_read, 0x40000000 /* SYS_read */ );
 
 ssize_t wld_write( int fd, const void *buffer, size_t len );
-SYSCALL_FUNC( wld_write, 0x40000000 + 1 /* SYS_write */ );
+SYSCALL_FUNC( wld_write, 0x40000001 /* SYS_write */ );
 
 int wld_open( const char *name, int flags );
-SYSCALL_FUNC( wld_open, 0x40000000 + 2 /* SYS_open */ );
+SYSCALL_FUNC( wld_open, 0x40000002 /* SYS_open */ );
 
 int wld_close( int fd );
-SYSCALL_FUNC( wld_close, 0x40000000 + 3 /* SYS_close */ );
+SYSCALL_FUNC( wld_close, 0x40000003 /* SYS_close */ );
 
 void *wld_mmap( void *start, size_t len, int prot, int flags, int fd, off_t offset );
-SYSCALL_FUNC( wld_mmap, 0x40000000 + 9 /* SYS_mmap */ );
+SYSCALL_FUNC( wld_mmap, 0x40000009 /* SYS_mmap */ );
 
 int wld_mprotect( const void *addr, size_t len, int prot );
-SYSCALL_FUNC( wld_mprotect, 0x40000000 + 10 /* SYS_mprotect */ );
+SYSCALL_FUNC( wld_mprotect, 0x4000000A /* SYS_mprotect */ );
 
 int wld_prctl( int code, int arg );
-SYSCALL_FUNC( wld_prctl, 0x40000000 + 157 /* SYS_prctl */ );
+SYSCALL_FUNC( wld_prctl, 0x4000009D /* SYS_prctl */ );
 
 uid_t wld_getuid(void);
-SYSCALL_NOERR( wld_getuid, 0x40000000 + 102 /* SYS_getuid */ );
+SYSCALL_NOERR( wld_getuid, 0x40000066 /* SYS_getuid */ );
 
 gid_t wld_getgid(void);
-SYSCALL_NOERR( wld_getgid, 0x40000000 + 104 /* SYS_getgid */ );
+SYSCALL_NOERR( wld_getgid, 0x40000068 /* SYS_getgid */ );
 
 uid_t wld_geteuid(void);
-SYSCALL_NOERR( wld_geteuid, 0x40000000 + 107 /* SYS_geteuid */ );
+SYSCALL_NOERR( wld_geteuid, 0x4000006B /* SYS_geteuid */ );
 
 gid_t wld_getegid(void);
-SYSCALL_NOERR( wld_getegid, 0x40000000 + 108 /* SYS_getegid */ );
+SYSCALL_NOERR( wld_getegid, 0x4000006C /* SYS_getegid */ );
 
 #else
 
