diff -urp wine-1.8.2.orig/include/basetsd.h wine-1.8.2/include/basetsd.h
--- wine-1.8.2.orig/include/basetsd.h	2016-04-12 04:42:53.000000000 -0700
+++ wine-1.8.2/include/basetsd.h	2016-10-22 21:25:14.506153961 -0700
@@ -58,7 +58,11 @@ extern "C" {
 #  endif
 #  ifndef __int64
 #    if defined(_WIN64) && !defined(__MINGW64__)
-#      define __int64 long
+#      if defined(__X32__)
+#        define __int64 long long
+#      else
+#        define __int64 long
+#      endif
 #    else
 #      define __int64 long long
 #    endif
diff -urp wine-1.8.2.orig/include/msvcrt/crtdefs.h wine-1.8.2/include/msvcrt/crtdefs.h
--- wine-1.8.2.orig/include/msvcrt/crtdefs.h	2016-04-12 04:42:53.000000000 -0700
+++ wine-1.8.2/include/msvcrt/crtdefs.h	2016-10-22 21:20:06.607412743 -0700
@@ -35,7 +35,11 @@
 
 #if !defined(_MSC_VER) && !defined(__int64)
 # if defined(_WIN64) && !defined(__MINGW64__)
-#   define __int64 long
+#   if defined(__X32__)
+#     define __int64 long long
+#   else
+#     define __int64 long
+#   endif
 # else
 #   define __int64 long long
 # endif
diff -urp wine-1.8.2.orig/include/wine/port.h wine-1.8.2/include/wine/port.h
--- wine-1.8.2.orig/include/wine/port.h	2016-04-12 04:42:53.000000000 -0700
+++ wine-1.8.2/include/wine/port.h	2016-10-22 21:18:45.473318826 -0700
@@ -58,7 +58,11 @@
 
 #if !defined(_MSC_VER) && !defined(__int64)
 #  if defined(__x86_64__) || defined(__aarch64__) || defined(_WIN64)
-#    define __int64 long
+#    if defined(__X32__)
+#      define __int64 long long
+#    else
+#      define __int64 long
+#    endif
 #  else
 #    define __int64 long long
 #  endif
