diff -ur wine-1.8.2.orig/include/basetsd.h wine-1.8.2/include/basetsd.h
--- wine-1.8.2.orig/include/basetsd.h	2016-09-01 02:59:07.344823602 -0700
+++ wine-1.8.2/include/basetsd.h	2016-09-01 03:18:26.318805049 -0700
@@ -36,7 +36,7 @@
  * 64-bit.
  */
 
-#if ((defined(__x86_64__) && !defined(__X32__)) || defined(__powerpc64__) || defined(__sparc64__) || defined(__aarch64__)) && !defined(_WIN64)
+#if ((defined(__x86_64__) || defined(__X32__)) || defined(__powerpc64__) || defined(__sparc64__) || defined(__aarch64__)) && !defined(_WIN64)
 #define _WIN64
 #endif
 
@@ -57,7 +57,7 @@
 #    define __int32 int
 #  endif
 #  ifndef __int64
-#    if defined(_WIN64) && !defined(__MINGW64__)
+#    if defined(_WIN64) && !defined(__MINGW64__) && !defined(__X32__)
 #      define __int64 long
 #    else
 #      define __int64 long long
@@ -119,7 +119,7 @@
 typedef /* [public] */ unsigned __int3264 ULONG_PTR, *PULONG_PTR;
 typedef ULONG_PTR                   DWORD_PTR, *PDWORD_PTR;
 
-#elif defined(_WIN64)
+#elif defined(_WIN64) && !defined(__X32__)
 
 #define __int3264 __int64
 
diff -ur wine-1.8.2.orig/include/msvcrt/crtdefs.h wine-1.8.2/include/msvcrt/crtdefs.h
--- wine-1.8.2.orig/include/msvcrt/crtdefs.h	2016-09-01 02:59:07.336823602 -0700
+++ wine-1.8.2/include/msvcrt/crtdefs.h	2016-09-01 03:09:10.940813939 -0700
@@ -29,12 +29,12 @@
 # error You cannot use both wine/port.h and msvcrt headers
 #endif
 
-#if ((defined(__x86_64__) && !defined(__X32__)) || defined(__powerpc64__) || defined(__sparc64__) || defined(__aarch64__)) && !defined(_WIN64)
+#if ((defined(__x86_64__) || defined(__X32__)) || defined(__powerpc64__) || defined(__sparc64__) || defined(__aarch64__)) && !defined(_WIN64)
 #define _WIN64
 #endif
 
 #if !defined(_MSC_VER) && !defined(__int64)
-# if defined(_WIN64) && !defined(__MINGW64__)
+# if defined(_WIN64) && !defined(__MINGW64__) && !defined(__X32__)
 #   define __int64 long
 # else
 #   define __int64 long long
diff -ur wine-1.8.2.orig/include/windef.h wine-1.8.2/include/windef.h
--- wine-1.8.2.orig/include/windef.h	2016-09-01 02:59:07.336823602 -0700
+++ wine-1.8.2/include/windef.h	2016-09-01 03:10:16.568812889 -0700
@@ -37,7 +37,7 @@
 
 /* Calling conventions definitions */
 
-#if ((defined(__x86_64__) && !defined(__X32__)) || defined(__powerpc64__) || defined(__sparc64__) || defined(__aarch64__)) && !defined(_WIN64)
+#if ((defined(__x86_64__) || defined(__X32__)) || defined(__powerpc64__) || defined(__sparc64__) || defined(__aarch64__)) && !defined(_WIN64)
 #define _WIN64
 #endif
 
diff -ur wine-1.8.2.orig/include/wine/port.h wine-1.8.2/include/wine/port.h
--- wine-1.8.2.orig/include/wine/port.h	2016-09-01 02:59:07.346823602 -0700
+++ wine-1.8.2/include/wine/port.h	2016-09-01 03:26:02.581797745 -0700
@@ -57,7 +57,7 @@
  */
 
 #if !defined(_MSC_VER) && !defined(__int64)
-#  if (defined(__x86_64__) && !defined(__X32__)) || defined(__aarch64__) || defined(_WIN64)
+#  if (defined(__x86_64__) || defined(__aarch64__) || defined(_WIN64)) && !defined(__X32__)
 #    define __int64 long
 #  else
 #    define __int64 long long
diff -ur wine-1.8.2.orig/include/wine/test.h wine-1.8.2/include/wine/test.h
--- wine-1.8.2.orig/include/wine/test.h	2016-04-12 04:42:53.000000000 -0700
+++ wine-1.8.2/include/wine/test.h	2016-09-01 03:54:06.266770792 -0700
@@ -85,7 +85,7 @@
 #define START_TEST(name) void func_##name(void)
 #endif
 
-#if defined(__x86_64__) && defined(__GNUC__) && defined(__WINE_USE_MSVCRT)
+#if (defined(__x86_64__) && !defined(__X32__)) && defined(__GNUC__) && defined(__WINE_USE_MSVCRT)
 #define __winetest_cdecl __cdecl
 #define __winetest_va_list __builtin_ms_va_list
 #else
