diff -urp wine-1.9.9.orig/include/msvcrt/crtdefs.h wine-1.9.9/include/msvcrt/crtdefs.h
--- wine-1.9.9.orig/include/msvcrt/crtdefs.h	2016-10-25 02:56:19.782345508 -0700
+++ wine-1.9.9/include/msvcrt/crtdefs.h	2016-10-25 02:59:56.940580039 -0700
@@ -59,7 +59,9 @@
 #   error You need to define __stdcall for your compiler
 #  endif
 # elif defined(__x86_64__) && defined (__GNUC__)
-#  if (__GNUC__ > 5) || ((__GNUC__ == 5) && (__GNUC_MINOR__ >= 3))
+#  if defined(__X32__)
+#   define __stdcall
+#  elif (__GNUC__ > 5) || ((__GNUC__ == 5) && (__GNUC_MINOR__ >= 3))
 #   define __stdcall __attribute__((ms_abi)) __attribute__((__force_align_arg_pointer__))
 #  else
 #   define __stdcall __attribute__((ms_abi))
@@ -77,7 +79,9 @@
 #   define __cdecl __attribute__((__cdecl__))
 #  endif
 # elif defined(__x86_64__) && defined (__GNUC__)
-#  if (__GNUC__ > 5) || ((__GNUC__ == 5) && (__GNUC_MINOR__ >= 3))
+#  if defined(__X32__)
+#   define __cdecl
+#  elif (__GNUC__ > 5) || ((__GNUC__ == 5) && (__GNUC_MINOR__ >= 3))
 #   define __cdecl __attribute__((ms_abi)) __attribute__((__force_align_arg_pointer__))
 #  else
 #   define __cdecl __attribute__((ms_abi))
@@ -88,7 +92,7 @@
 #endif /* __cdecl */
 
 #ifndef __ms_va_list
-# if defined(__x86_64__) && defined (__GNUC__)
+# if (defined(__x86_64__) && !defined(__X32__)) && defined (__GNUC__)
 #  define __ms_va_list __builtin_ms_va_list
 #  define __ms_va_start(list,arg) __builtin_ms_va_start(list,arg)
 #  define __ms_va_end(list) __builtin_ms_va_end(list)
Only in wine-1.9.9/include/msvcrt: crtdefs.h.orig
Only in wine-1.9.9/include/msvcrt: crtdefs.h.rej
diff -urp wine-1.9.9.orig/include/windef.h wine-1.9.9/include/windef.h
--- wine-1.9.9.orig/include/windef.h	2016-04-29 09:04:00.000000000 -0700
+++ wine-1.9.9/include/windef.h	2016-10-25 03:01:25.818950263 -0700
@@ -37,7 +37,7 @@ extern "C" {
 
 /* Calling conventions definitions */
 
-#if (defined(__x86_64__) || defined(__powerpc64__) || defined(__sparc64__) || defined(__aarch64__)) && !defined(_WIN64)
+#if ((defined(__x86_64__) && !defined(__X32__)) || defined(__powerpc64__) || defined(__sparc64__) || defined(__aarch64__)) && !defined(_WIN64)
 #define _WIN64
 #endif
 
@@ -64,7 +64,9 @@ extern "C" {
 #   error You need to define __stdcall for your compiler
 #  endif
 # elif defined(__x86_64__) && defined (__GNUC__)
-#  if (__GNUC__ > 5) || ((__GNUC__ == 5) && (__GNUC_MINOR__ >= 3))
+#  if defined(__X32__)
+#   define __stdcall
+#  elif (__GNUC__ > 5) || ((__GNUC__ == 5) && (__GNUC_MINOR__ >= 3))
 #   define __stdcall __attribute__((ms_abi)) __attribute__((__force_align_arg_pointer__))
 #  else
 #   define __stdcall __attribute__((ms_abi))
@@ -82,7 +84,9 @@ extern "C" {
 #   define __cdecl __attribute__((__cdecl__))
 #  endif
 # elif defined(__x86_64__) && defined (__GNUC__)
-#  if (__GNUC__ > 5) || ((__GNUC__ == 5) && (__GNUC_MINOR__ >= 3))
+#  if defined(__X32__)
+#   define __cdecl
+#  elif (__GNUC__ > 5) || ((__GNUC__ == 5) && (__GNUC_MINOR__ >= 3))
 #   define __cdecl __attribute__((ms_abi)) __attribute__((__force_align_arg_pointer__))
 #  else
 #   define __cdecl __attribute__((ms_abi))
@@ -93,7 +97,7 @@ extern "C" {
 #endif /* __cdecl */
 
 #ifndef __ms_va_list
-# if defined(__x86_64__) && defined (__GNUC__)
+# if (defined(__x86_64__) && !defined(__X32__)) && defined (__GNUC__)
 #  define __ms_va_list __builtin_ms_va_list
 #  define __ms_va_start(list,arg) __builtin_ms_va_start(list,arg)
 #  define __ms_va_end(list) __builtin_ms_va_end(list)
Only in wine-1.9.9/include: windef.h.orig
Only in wine-1.9.9/include: windef.h.rej
