diff -ur wine-1.8.2.orig/include/basetsd.h wine-1.8.2/include/basetsd.h
--- wine-1.8.2.orig/include/basetsd.h	2016-04-12 04:42:53.000000000 -0700
+++ wine-1.8.2/include/basetsd.h	2016-08-29 15:15:07.603477714 -0700
@@ -36,7 +36,7 @@
  * 64-bit.
  */
 
-#if (defined(__x86_64__) || defined(__powerpc64__) || defined(__sparc64__) || defined(__aarch64__)) && !defined(_WIN64)
+#if ((defined(__x86_64__) && !defined(__X32__)) || defined(__powerpc64__) || defined(__sparc64__) || defined(__aarch64__)) && !defined(_WIN64)
 #define _WIN64
 #endif
 
@@ -93,12 +93,20 @@
 typedef unsigned __int64 ULONG64, *PULONG64;
 typedef unsigned __int64 DWORD64, *PDWORD64;
 #else
+#if defined(__X32__)
+typedef signed __int64   DECLSPEC_ALIGN(4) INT64, *PINT64;
+typedef unsigned __int64 DECLSPEC_ALIGN(4) UINT64, *PUINT64;
+typedef signed __int64   DECLSPEC_ALIGN(4) LONG64, *PLONG64;
+typedef unsigned __int64 DECLSPEC_ALIGN(4) ULONG64, *PULONG64;
+typedef unsigned __int64 DECLSPEC_ALIGN(4) DWORD64, *PDWORD64;
+#else
 typedef signed __int64   DECLSPEC_ALIGN(8) INT64, *PINT64;
 typedef unsigned __int64 DECLSPEC_ALIGN(8) UINT64, *PUINT64;
 typedef signed __int64   DECLSPEC_ALIGN(8) LONG64, *PLONG64;
 typedef unsigned __int64 DECLSPEC_ALIGN(8) ULONG64, *PULONG64;
 typedef unsigned __int64 DECLSPEC_ALIGN(8) DWORD64, *PDWORD64;
 #endif
+#endif
 
 /* Basic pointer-sized integer types */
 
diff -ur wine-1.8.2.orig/include/msvcrt/crtdefs.h wine-1.8.2/include/msvcrt/crtdefs.h
--- wine-1.8.2.orig/include/msvcrt/crtdefs.h	2016-04-12 04:42:53.000000000 -0700
+++ wine-1.8.2/include/msvcrt/crtdefs.h	2016-08-29 15:17:27.980475467 -0700
@@ -29,7 +29,7 @@
 # error You cannot use both wine/port.h and msvcrt headers
 #endif
 
-#if (defined(__x86_64__) || defined(__powerpc64__) || defined(__sparc64__) || defined(__aarch64__)) && !defined(_WIN64)
+#if ((defined(__x86_64__) && !defined(__X32__)) || defined(__powerpc64__) || defined(__sparc64__) || defined(__aarch64__)) && !defined(_WIN64)
 #define _WIN64
 #endif
 
@@ -55,7 +55,11 @@
 #   error You need to define __stdcall for your compiler
 #  endif
 # elif defined(__x86_64__) && defined (__GNUC__)
-#  define __stdcall __attribute__((ms_abi))
+#  if defined(__X32__)
+#    define __stdcall
+#  else
+#    define __stdcall __attribute__((ms_abi))
+#  endif
 # else
 #  define __stdcall
 # endif
@@ -69,14 +73,18 @@
 #   define __cdecl __attribute__((__cdecl__))
 #  endif
 # elif defined(__x86_64__) && defined (__GNUC__)
-#  define __cdecl __attribute__((ms_abi))
+#  if defined(__X32__)
+#    define __cdecl
+#  else
+#    define __cdecl __attribute__((ms_abi))
+#  endif
 # elif !defined(_MSC_VER)
 #  define __cdecl
 # endif
 #endif /* __cdecl */
 
 #ifndef __ms_va_list
-# if defined(__x86_64__) && defined (__GNUC__)
+# if (defined(__x86_64__) && !defined(__X32__)) && defined (__GNUC__)
 #  define __ms_va_list __builtin_ms_va_list
 #  define __ms_va_start(list,arg) __builtin_ms_va_start(list,arg)
 #  define __ms_va_end(list) __builtin_ms_va_end(list)
diff -ur wine-1.8.2.orig/include/windef.h wine-1.8.2/include/windef.h
--- wine-1.8.2.orig/include/windef.h	2016-04-12 04:42:53.000000000 -0700
+++ wine-1.8.2/include/windef.h	2016-08-29 15:18:05.994474858 -0700
@@ -37,7 +37,7 @@
 
 /* Calling conventions definitions */
 
-#if (defined(__x86_64__) || defined(__powerpc64__) || defined(__sparc64__) || defined(__aarch64__)) && !defined(_WIN64)
+#if ((defined(__x86_64__) && !defined(__X32__)) || defined(__powerpc64__) || defined(__sparc64__) || defined(__aarch64__)) && !defined(_WIN64)
 #define _WIN64
 #endif
 
@@ -64,7 +64,11 @@
 #   error You need to define __stdcall for your compiler
 #  endif
 # elif defined(__x86_64__) && defined (__GNUC__)
-#  define __stdcall __attribute__((ms_abi))
+#  if defined(__X32__)
+#    define __stdcall
+#  else
+#    define __stdcall __attribute__((ms_abi))
+#  endif
 # else  /* __i386__ */
 #  define __stdcall
 # endif  /* __i386__ */
@@ -78,14 +82,18 @@
 #   define __cdecl __attribute__((__cdecl__))
 #  endif
 # elif defined(__x86_64__) && defined (__GNUC__)
-#  define __cdecl __attribute__((ms_abi))
+#  if defined(__X32__)
+#    define __cdecl
+#  else
+#    define __cdecl __attribute__((ms_abi))
+#  endif
 # elif !defined(_MSC_VER)
 #  define __cdecl
 # endif
 #endif /* __cdecl */
 
 #ifndef __ms_va_list
-# if defined(__x86_64__) && defined (__GNUC__)
+# if (defined(__x86_64__) && !defined(__X32__)) && defined (__GNUC__)
 #  define __ms_va_list __builtin_ms_va_list
 #  define __ms_va_start(list,arg) __builtin_ms_va_start(list,arg)
 #  define __ms_va_end(list) __builtin_ms_va_end(list)
diff -ur wine-1.8.2.orig/include/winnt.h wine-1.8.2/include/winnt.h
--- wine-1.8.2.orig/include/winnt.h	2016-04-12 04:42:53.000000000 -0700
+++ wine-1.8.2/include/winnt.h	2016-08-29 15:16:10.098476713 -0700
@@ -460,8 +460,13 @@
 typedef signed __int64   LONGLONG,  *PLONGLONG;
 typedef unsigned __int64 ULONGLONG, *PULONGLONG;
 # else
+#if defined(__X32__)
+typedef signed __int64   DECLSPEC_ALIGN(4) LONGLONG,   *PLONGLONG;
+typedef unsigned __int64 DECLSPEC_ALIGN(4) ULONGLONG,  *PULONGLONG;
+#else
 typedef signed __int64   DECLSPEC_ALIGN(8) LONGLONG,   *PLONGLONG;
 typedef unsigned __int64 DECLSPEC_ALIGN(8) ULONGLONG,  *PULONGLONG;
+#endif
 # endif
 #endif
 
