diff -urp wine-1.8.2.orig/include/winbase.h wine-1.8.2/include/winbase.h
--- wine-1.8.2.orig/include/winbase.h	2016-10-22 20:21:53.125426990 -0700
+++ wine-1.8.2/include/winbase.h	2016-10-22 21:57:26.270823672 -0700
@@ -2734,6 +2734,14 @@ extern WCHAR * CDECL wine_get_dos_file_n
 
 /* Interlocked functions */
 
+#if defined(__x86_64__)
+#if defined(__X32__)
+#define PTR_OPSIZE "l"
+#else
+#define PTR_OPSIZE "q"
+#endif
+#endif
+
 #ifdef __i386__
 # if defined(__GNUC__) && !defined(_NTSYSTEM_) && ((__GNUC__ > 3) || ((__GNUC__ == 3) && (__GNUC_MINOR__ >= 2)))
 
@@ -2950,7 +2958,7 @@ static FORCEINLINE HANDLE WINAPI GetProc
 {
     HANDLE *pdb;
 #ifdef __x86_64__
-    __asm__ __volatile__( ".byte 0x65\n\tmovq 0x60,%0" : "=r" (pdb) );
+    __asm__ __volatile__( ".byte 0x65\n\tmov"PTR_OPSIZE" 0x60,%0" : "=r" (pdb) );
     return pdb[0x30 / sizeof(HANDLE)];  /* get dword at offset 0x30 in pdb */
 #else
     __asm__ __volatile__( ".byte 0x64\n\tmovl 0x30,%0" : "=r" (pdb) );
diff -urp wine-1.8.2.orig/include/winnt.h wine-1.8.2/include/winnt.h
--- wine-1.8.2.orig/include/winnt.h	2016-10-22 21:20:58.898406775 -0700
+++ wine-1.8.2/include/winnt.h	2016-10-22 21:58:22.796981369 -0700
@@ -2317,6 +2317,14 @@ typedef struct _NT_TIB
 
 struct _TEB;
 
+#if defined(__x86_64__)
+#if defined(__X32__)
+#define PTR_OPSIZE "l"
+#else
+#define PTR_OPSIZE "q"
+#endif
+#endif
+
 #if defined(__i386__) && defined(__GNUC__) && ((__GNUC__ > 3) || ((__GNUC__ == 3) && (__GNUC_MINOR__ >= 2)))
 static FORCEINLINE struct _TEB * WINAPI NtCurrentTeb(void)
 {
@@ -2336,7 +2344,7 @@ static FORCEINLINE struct _TEB * WINAPI
 static FORCEINLINE struct _TEB * WINAPI NtCurrentTeb(void)
 {
     struct _TEB *teb;
-    __asm__(".byte 0x65\n\tmovq (0x30),%0" : "=r" (teb));
+    __asm__(".byte 0x65\n\tmov"PTR_OPSIZE" (0x30),%0" : "=r" (teb));
     return teb;
 }
 #elif defined(__x86_64__) && defined(_MSC_VER)
