diff -ur wine-1.8.2.orig/include/wine/test.h wine-1.8.2/include/wine/test.h
--- wine-1.8.2.orig/include/wine/test.h	2016-04-12 04:42:53.000000000 -0700
+++ wine-1.8.2/include/wine/test.h	2016-09-01 03:54:06.266770792 -0700
@@ -85,7 +85,7 @@
 #define START_TEST(name) void func_##name(void)
 #endif
 
-#if defined(__x86_64__) && defined(__GNUC__) && defined(__WINE_USE_MSVCRT)
+#if (defined(__x86_64__) && !defined(__X32__)) && defined(__GNUC__) && defined(__WINE_USE_MSVCRT)
 #define __winetest_cdecl __cdecl
 #define __winetest_va_list __builtin_ms_va_list
 #else
