diff -ur wine-1.8.2.orig/loader/preloader.c wine-1.8.2/loader/preloader.c
--- wine-1.8.2.orig/loader/preloader.c	2016-04-12 04:42:53.000000000 -0700
+++ wine-1.8.2/loader/preloader.c	2016-08-29 16:40:18.253395902 -0700
@@ -108,7 +108,7 @@
 
 static struct wine_preload_info preload_info[] =
 {
-#ifdef __i386__
+#if defined(__i386__) || defined(__X32__)
     { (void *)0x00000000, 0x00010000 },  /* low 64k */
     { (void *)0x00010000, 0x00100000 },  /* DOS area */
     { (void *)0x00110000, 0x67ef0000 },  /* low memory area */
diff -ur wine-1.8.2.orig/tools/winebuild/main.c wine-1.8.2/tools/winebuild/main.c
--- wine-1.8.2.orig/tools/winebuild/main.c	2016-08-29 15:22:35.306470547 -0700
+++ wine-1.8.2/tools/winebuild/main.c	2016-08-29 16:31:26.970404407 -0700
@@ -545,10 +545,20 @@
     switch (target_cpu)
     {
     case CPU_x86:
+#if defined(__X32__)
+        if (force_pointer_size == 4) target_cpu = CPU_x86_64;
+        if (force_pointer_size == 8) target_cpu = CPU_x86_64;
+#else
         if (force_pointer_size == 8) target_cpu = CPU_x86_64;
+#endif
         break;
     case CPU_x86_64:
+#if defined(__X32__)
+        if (force_pointer_size == 4) target_cpu = CPU_x86_64;
+        if (force_pointer_size == 8) target_cpu = CPU_x86_64;
+#else
         if (force_pointer_size == 4) target_cpu = CPU_x86;
+#endif
         break;
     default:
         if (force_pointer_size == 8)
