diff -urp llvm-3.5.0.src.orig/projects/compiler-rt/lib/builtins/Makefile.mk llvm-3.5.0.src/projects/compiler-rt/lib/builtins/Makefile.mk
--- llvm-3.5.0.src.orig/projects/compiler-rt/lib/builtins/Makefile.mk	2014-07-10 03:06:42.000000000 -0700
+++ llvm-3.5.0.src/projects/compiler-rt/lib/builtins/Makefile.mk	2016-10-18 17:54:44.921055731 -0700
@@ -13,8 +13,11 @@ SubDirs :=
 # Add arch specific optimized implementations.
 SubDirs += i386 ppc x86_64 arm armv6m
 
+# Does not work in muslx32
+ExcludedSources := atomic.c gcc_personality_v0.c
+
 # Define the variables for this specific directory.
-Sources := $(foreach file,$(wildcard $(Dir)/*.c),$(notdir $(file)))
+Sources := $(foreach file,$(wildcard $(Dir)/*.c),$(filter-out $(ExcludedSources),$(notdir $(file))))
 ObjNames := $(Sources:%.c=%.o)
 Implementation := Generic
 
