diff -ur llvm-3.5.0.src.orig/projects/compiler-rt/lib/profile/InstrProfilingFile.c llvm-3.5.0.src/projects/compiler-rt/lib/profile/InstrProfilingFile.c
--- llvm-3.5.0.src.orig/projects/compiler-rt/lib/profile/InstrProfilingFile.c	2014-06-15 07:01:18.000000000 -0700
+++ llvm-3.5.0.src/projects/compiler-rt/lib/profile/InstrProfilingFile.c	2016-10-18 10:30:24.745346781 -0700
@@ -12,7 +12,7 @@
 #include <stdlib.h>
 #include <string.h>
 
-#define UNCONST(ptr) ((void *)(uintptr_t)(ptr))
+#define UNCONST(ptr) ((void *)(long)(ptr))
 
 static int writeFile(FILE *File) {
   /* Match logic in __llvm_profile_write_buffer(). */
