diff -urp llvm-3.5.0.src.orig/tools/clang/runtime/compiler-rt/Makefile llvm-3.5.0.src/tools/clang/runtime/compiler-rt/Makefile
--- llvm-3.5.0.src.orig/tools/clang/runtime/compiler-rt/Makefile	2016-10-18 18:51:49.851254260 -0700
+++ llvm-3.5.0.src/tools/clang/runtime/compiler-rt/Makefile	2016-10-18 19:03:38.343283617 -0700
@@ -177,9 +177,9 @@ RuntimeLibrary.linux.Configs += $(Runtim
 # We check that Clang can produce working 32/64-bit binaries by compiling a simple
 # executable.
 test_source = $(LLVM_SRC_ROOT)/tools/clang/runtime/compiler-rt/clang_linux_test_input.c
-ifeq ($(call TryCompile,$(ToolDir)/clang,$(test_source),-m64),0)
-RuntimeLibrary.linux.Configs += $(Runtime64BitConfigs)
-endif
+#ifeq ($(call TryCompile,$(ToolDir)/clang,$(test_source),-m64),0)
+#RuntimeLibrary.linux.Configs += $(Runtime64BitConfigs)
+#endif
 ifeq ($(call TryCompile,$(ToolDir)/clang,$(test_source),-m32),0)
 RuntimeLibrary.linux.Configs += $(Runtimex86Configs)
 endif
