diff -urp llvm-3.5.0.src.orig/lib/Support/Triple.cpp llvm-3.5.0.src/lib/Support/Triple.cpp
--- llvm-3.5.0.src.orig/lib/Support/Triple.cpp	2016-10-18 17:37:23.223818184 -0700
+++ llvm-3.5.0.src/lib/Support/Triple.cpp	2016-10-18 17:38:09.658567237 -0700
@@ -348,10 +348,10 @@ static Triple::EnvironmentType parseEnvi
     .StartsWith("gnux32", Triple::GNUX32)
     .StartsWith("code16", Triple::CODE16)
     .StartsWith("gnu", Triple::GNU)
-    .StartsWith("musl", Triple::Musl)
     .StartsWith("muslx32", Triple::MuslX32)
-    .StartsWith("musleabi", Triple::MuslEABI)
     .StartsWith("musleabihf", Triple::MuslEABIHF)
+    .StartsWith("musleabi", Triple::MuslEABI)
+    .StartsWith("musl", Triple::Musl)
     .StartsWith("android", Triple::Android)
     .StartsWith("msvc", Triple::MSVC)
     .StartsWith("itanium", Triple::Itanium)
