diff -urp llvm-3.7.1.src.orig/include/llvm/ADT/Triple.h llvm-3.7.1.src/include/llvm/ADT/Triple.h
--- llvm-3.7.1.src.orig/include/llvm/ADT/Triple.h	2015-07-06 16:58:14.000000000 -0700
+++ llvm-3.7.1.src/include/llvm/ADT/Triple.h	2016-10-12 22:20:45.116585434 -0700
@@ -167,6 +167,10 @@ public:
     EABIHF,
     Android,
 
+    Musl,
+    MuslEABI,
+    MuslEABIHF,
+
     MSVC,
     Itanium,
     Cygnus,
diff -urp llvm-3.7.1.src.orig/lib/Support/Triple.cpp llvm-3.7.1.src/lib/Support/Triple.cpp
--- llvm-3.7.1.src.orig/lib/Support/Triple.cpp	2015-07-06 16:58:14.000000000 -0700
+++ llvm-3.7.1.src/lib/Support/Triple.cpp	2016-10-12 22:20:47.403672647 -0700
@@ -192,6 +192,9 @@ const char *Triple::getEnvironmentTypeNa
   case CODE16: return "code16";
   case EABI: return "eabi";
   case EABIHF: return "eabihf";
+  case Musl: return "musl";
+  case MuslEABI: return "musleabi";
+  case MuslEABIHF: return "musleabihf";
   case Android: return "android";
   case MSVC: return "msvc";
   case Itanium: return "itanium";
@@ -426,6 +429,9 @@ static Triple::EnvironmentType parseEnvi
     .StartsWith("gnux32", Triple::GNUX32)
     .StartsWith("code16", Triple::CODE16)
     .StartsWith("gnu", Triple::GNU)
+    .StartsWith("musl", Triple::Musl)
+    .StartsWith("musleabi", Triple::MuslEABI)
+    .StartsWith("musleabihf", Triple::MuslEABIHF)
     .StartsWith("android", Triple::Android)
     .StartsWith("msvc", Triple::MSVC)
     .StartsWith("itanium", Triple::Itanium)
diff -urp llvm-3.7.1.src.orig/lib/Target/ARM/ARMSubtarget.h llvm-3.7.1.src/lib/Target/ARM/ARMSubtarget.h
--- llvm-3.7.1.src.orig/lib/Target/ARM/ARMSubtarget.h	2015-07-06 23:54:42.000000000 -0700
+++ llvm-3.7.1.src/lib/Target/ARM/ARMSubtarget.h	2016-10-12 22:20:49.483751965 -0700
@@ -381,8 +381,10 @@ public:
   bool isTargetEHABICompatible() const {
     return (TargetTriple.getEnvironment() == Triple::EABI ||
             TargetTriple.getEnvironment() == Triple::GNUEABI ||
+            TargetTriple.getEnvironment() == Triple::MuslEABI ||
             TargetTriple.getEnvironment() == Triple::EABIHF ||
             TargetTriple.getEnvironment() == Triple::GNUEABIHF ||
+            TargetTriple.getEnvironment() == Triple::MuslEABIHF ||
             TargetTriple.getEnvironment() == Triple::Android) &&
            !isTargetDarwin() && !isTargetWindows();
   }
diff -urp llvm-3.7.1.src.orig/lib/Target/ARM/ARMTargetMachine.cpp llvm-3.7.1.src/lib/Target/ARM/ARMTargetMachine.cpp
--- llvm-3.7.1.src.orig/lib/Target/ARM/ARMTargetMachine.cpp	2015-07-06 09:33:18.000000000 -0700
+++ llvm-3.7.1.src/lib/Target/ARM/ARMTargetMachine.cpp	2016-10-12 22:27:52.356876096 -0700
@@ -95,6 +95,8 @@ computeTargetABI(const Triple &TT, Strin
     case llvm::Triple::Android:
     case llvm::Triple::GNUEABI:
     case llvm::Triple::GNUEABIHF:
+    case llvm::Triple::MuslEABI:
+    case llvm::Triple::MuslEABIHF:
     case llvm::Triple::EABIHF:
     case llvm::Triple::EABI:
       TargetABI = ARMBaseTargetMachine::ARM_ABI_AAPCS;
