diff -ur xterm-320.orig/main.c xterm-320/main.c
--- xterm-320.orig/main.c	2015-08-26 16:39:36.000000000 -0700
+++ xterm-320/main.c	2016-07-30 00:10:08.733660006 -0700
@@ -2628,7 +2628,7 @@
 
 #if defined(USE_OPENPTY)
     result = openpty(pty, &opened_tty, ttydev, NULL, NULL);
-#elif defined(HAVE_POSIX_OPENPT) && defined(HAVE_PTSNAME) && defined(HAVE_GRANTPT_PTY_ISATTY)
+#elif defined(HAVE_POSIX_OPENPT) && defined(HAVE_PTSNAME)
     if ((*pty = posix_openpt(O_RDWR)) >= 0) {
 	char *name = ptsname(*pty);
 	if (name != 0) {
Binary files xterm-320.orig/main.o and xterm-320/main.o differ
Binary files xterm-320.orig/xterm and xterm-320/xterm differ
