diff -ur gcc-4.9.3.orig/gcc/config/i386/i386.c gcc-4.9.3/gcc/config/i386/i386.c
--- gcc-4.9.3.orig/gcc/config/i386/i386.c	2015-06-10 02:26:06.000000000 -0700
+++ gcc-4.9.3/gcc/config/i386/i386.c	2016-08-20 23:44:36.877436196 -0700
@@ -5988,18 +5988,7 @@
       if (abi == SYSV_ABI)
 	{
 	  if (lookup_attribute ("ms_abi", TYPE_ATTRIBUTES (fntype)))
-	    {
-	      if (TARGET_X32)
-		{
-		  static bool warned = false;
-		  if (!warned)
-		    {
-		      error ("X32 does not support ms_abi attribute");
-		      warned = true;
-		    }
-		}
 	      abi = MS_ABI;
-	    }
 	}
       else if (lookup_attribute ("sysv_abi", TYPE_ATTRIBUTES (fntype)))
 	abi = SYSV_ABI;
