--- a/gcc-7.3.0/libstdc++-v3/include/c_global/cmath.orig	2017-01-09 17:15:58.932108000 +0000
+++ b/gcc-7.3.0/libstdc++-v3/include/c_global/cmath	2018-04-24 04:25:13.357212425 +0000
@@ -42,7 +42,7 @@
 #include <bits/cpp_type_traits.h>
 #include <ext/type_traits.h>
 #define _GLIBCXX_INCLUDE_NEXT_C_HEADERS
-#include_next <math.h>
+#include <math.h>
 #undef _GLIBCXX_INCLUDE_NEXT_C_HEADERS
 #include <bits/std_abs.h>
 
