--- ./gcc-7.3.0/libstdc++-v3/Makefile.in.orig	2018-04-18 00:35:06.888245931 +0000
+++ ./gcc-7.3.0/libstdc++-v3/Makefile.in	2018-04-18 02:18:41.577310590 +0000
@@ -138,6 +138,7 @@ cc_for_target4 = @CC@
 cxx4 = @CXX@
 cxxcpp4 = @CXXCPP@
 ldflags1 = @LDFLAGS@
+cxxflags1 = @CXXFLAGS@
 ifneq (,$(findstring m64,$(CFLAGS)))
 cc_for_target3 = $(subst /x86_64-pc-linux-muslx32/usr/lib,/x86_64-pc-linux-musl/usr/lib,$(cc_for_target4))
 cc_for_target2 = $(subst /x86_64-pc-linux-muslx32/usr/bin,/x86_64-pc-linux-musl/usr/bin,$(cc_for_target3))
@@ -159,6 +160,7 @@ cxxcpp2 = $(subst /x86_64-pc-linux-muslx
 cxxcpp1 = $(subst /x86_64-pc-linux-muslx32/usr/include,/x86_64-pc-linux-musl/usr/include,$(cxxcpp2))
 cxxcpp = $(subst /x86_64-pc-linux-muslx32/usr/sys-include,/x86_64-pc-linux-musl/usr/sys-include,$(cxxcpp1))
 cxxcpp += --sysroot=/usr/x86_64-pc-linux-musl/usr/lib/
+cxxflags = $(cxxflags1) -B/usr/x86_64-pc-linux-musl/usr/bin/ -B/usr/x86_64-pc-linux-musl/usr/lib/ -I /usr/x86_64-pc-linux-musl/usr/include -I /usr/x86_64-pc-linux-musl/usr/sys-include
 ldflags = $(ldflags1) -Wl,-melf_x86_64
 else ifneq (,$(findstring mx32,$(CFLAGS)))
 cc_for_target3 = $(subst /x86_64-pc-linux-muslx32/usr/lib,/x86_64-pc-linux-muslx32/usr/lib,$(cc_for_target4))
@@ -181,6 +183,7 @@ cxxcpp2 = $(subst /x86_64-pc-linux-muslx
 cxxcpp1 = $(subst /x86_64-pc-linux-muslx32/usr/include,/x86_64-pc-linux-muslx32/usr/include,$(cxxcpp2))
 cxxcpp = $(subst /x86_64-pc-linux-muslx32/usr/sys-include,/x86_64-pc-linux-muslx32/usr/sys-include,$(cxxcpp1))
 cxxcpp += --sysroot=/usr/x86_64-pc-linux-muslx32/usr/lib/
+cxxflags = $(cxxflags1) -B/usr/x86_64-pc-linux-muslx32/usr/bin/ -B/usr/x86_64-pc-linux-muslx32/usr/lib/ -I /usr/x86_64-pc-linux-muslx32/usr/include -I /usr/x86_64-pc-linux-muslx32/usr/sys-include
 ldflags = $(ldflags1) -Wl,-melf32_x86_64
 else ifneq (,$(findstring m32,$(CFLAGS)))
 cc_for_target3 = $(subst /x86_64-pc-linux-muslx32/usr/lib,/i686-pc-linux-musl/usr/lib,$(cc_for_target4))
@@ -203,11 +206,13 @@ cxxcpp2 = $(subst /x86_64-pc-linux-muslx
 cxxcpp1 = $(subst /x86_64-pc-linux-muslx32/usr/include,/i686-pc-linux-musl/usr/include,$(cxxcpp2))
 cxxcpp = $(subst /x86_64-pc-linux-muslx32/usr/sys-include,/i686-pc-linux-musl/usr/sys-include,$(cxxcpp1))
 cxxcpp += --sysroot=/usr/i686-pc-linux-musl/usr/lib/
+cxxflags = $(cxxflags1) -B/usr/i686-pc-linux-musl/usr/bin/ -B/usr/i686-pc-linux-musl/usr/lib/ -I /usr/i686-pc-linux-musl/usr/include -I /usr/i686-pc-linux-musl/usr/sys-include
 ldflags = $(ldflags1) -Wl,-melf_i386
 else
 cc_for_target = $(cc_for_target4)
 cxx = $(cxx4)
 cxxcpp = $(cxxcpp4)
+cxxflags = $(cxxflags1) -B/usr/x86_64-pc-linux-muslx32/usr/bin/ -B/usr/x86_64-pc-linux-muslx32/usr/lib/ -I /usr/x86_64-pc-linux-muslx32/usr/include -I /usr/x86_64-pc-linux-muslx32/usr/sys-include
 ldflags = $(ldflags1)
 endif
 CC = $(cc_for_target)
@@ -233,7 +238,7 @@ CTIME_H = @CTIME_H@
 CXX = $(cxx)
 CXXCPP = $(cxxcpp)
 CXXFILT = @CXXFILT@
-CXXFLAGS = @CXXFLAGS@
+CXXFLAGS = $(cxxflags)
 CYGPATH_W = @CYGPATH_W@
 C_INCLUDE_DIR = @C_INCLUDE_DIR@
 DBLATEX = @DBLATEX@
