--- util-linux-2.30.2-r1.ebuild.orig	2020-11-28 23:00:55.000000000 -0800
+++ util-linux-2.30.2-r1.ebuild	2020-11-21 17:03:10.234951572 -0800
@@ -113,6 +113,10 @@ multilib_src_configure() {
 	export ac_cv_header_security_pam_misc_h=$(multilib_native_usex pam) #485486
 	export ac_cv_header_security_pam_appl_h=$(multilib_native_usex pam) #545042
 
+	if [[ "${CHOST}" =~ "muslx32" ]] ; then
+		append-cppflags -Dsigsetjmp=__sigsetjmp
+	fi
+
 	local myeconfargs=(
 		--disable-chfn-chsh
 		--disable-login
