diff -urp iputils-s20151218.orig/ping.c iputils-s20151218/ping.c
--- iputils-s20151218.orig/ping.c	2016-11-05 00:41:25.590816624 -0700
+++ iputils-s20151218/ping.c	2016-11-05 00:42:24.406064414 -0700
@@ -746,16 +746,7 @@ int send_probe()
 		icp->checksum = in_cksum((u_short *)&tmp_tv, sizeof(tmp_tv), ~icp->checksum);
 	}
 
-	do {
-		static struct iovec iov = {outpack, 0};
-		static struct msghdr m = { &whereto, sizeof(whereto),
-						   &iov, 1, &cmsg, 0, 0 };
-		m.msg_controllen = cmsg_len;
-		iov.iov_len = cc;
-
-		i = sendmsg(icmp_sock, &m, confirm);
-		confirm = 0;
-	} while (0);
+        i = sendto(icmp_sock, icp, cc, 0, (struct sockaddr*)&whereto, sizeof(whereto));
 
 	return (cc == i ? 0 : i);
 }
