diff -ur gstreamer-1.6.3.orig/gst/Makefile.in gstreamer-1.6.3/gst/Makefile.in
--- gstreamer-1.6.3.orig/gst/Makefile.in	2016-06-19 21:18:46.356315835 +0000
+++ gstreamer-1.6.3/gst/Makefile.in	2016-06-19 21:28:03.058298009 +0000
@@ -921,7 +921,7 @@
 @HAVE_INTROSPECTION_TRUE@gir_DATA = $(BUILT_GIRSOURCES)
 @HAVE_INTROSPECTION_TRUE@typelibsdir = $(libdir)/girepository-1.0/
 @HAVE_INTROSPECTION_TRUE@typelibs_DATA = $(BUILT_GIRSOURCES:.gir=.typelib)
-all: $(BUILT_SOURCES)
+all: libgstreamer-1.0.la $(BUILT_SOURCES)
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
 .SUFFIXES:
