diff -ur gstreamer-1.6.3.orig/gst/Makefile.in gstreamer-1.6.3/gst/Makefile.in
--- gstreamer-1.6.3.orig/gst/Makefile.in	2016-06-19 23:14:21.305093770 +0000
+++ gstreamer-1.6.3/gst/Makefile.in	2016-06-19 23:23:09.332076862 +0000
@@ -921,9 +921,13 @@
 @HAVE_INTROSPECTION_TRUE@gir_DATA = $(BUILT_GIRSOURCES)
 @HAVE_INTROSPECTION_TRUE@typelibsdir = $(libdir)/girepository-1.0/
 @HAVE_INTROSPECTION_TRUE@typelibs_DATA = $(BUILT_GIRSOURCES:.gir=.typelib)
-all: libgstreamer-1.0.la $(BUILT_SOURCES)
+all: sub libgstreamer-1.0.la $(BUILT_SOURCES)
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
+sub:
+	$(MAKE) -C parse
+	$(MAKE) -C printf
+
 .SUFFIXES:
 .SUFFIXES: .c .lo .o .obj
 $(srcdir)/Makefile.in: @MAINTAINER_MODE_TRUE@ $(srcdir)/Makefile.am  $(am__configure_deps)
