--- ffmpeg-2.8.6.orig/libavutil/x86/x86inc.asm	2016-01-31 16:52:22.000000000 -0800
+++ ffmpeg-2.8.6/libavutil/x86/x86inc.asm	2016-10-19 22:40:33.206871593 -0700
@@ -226,6 +226,12 @@ DECLARE_REG_TMP_SIZE 0,1,2,3,4,5,6,7,8,9
     %define gprsize 4
 %endif
 
+%if ARCH_X86_64_X64
+    %define ptrsize 8
+%else
+    %define ptrsize 4
+%endif
+
 %macro PUSH 1
     push %1
     %ifidn rstk, rsp
@@ -304,6 +310,7 @@ DECLARE_REG_TMP_SIZE 0,1,2,3,4,5,6,7,8,9
         %assign %%i 0
         %rep n_arg_names
             CAT_UNDEF arg_name %+ %%i, q
+            CAT_UNDEF arg_name %+ %%i, p
             CAT_UNDEF arg_name %+ %%i, d
             CAT_UNDEF arg_name %+ %%i, w
             CAT_UNDEF arg_name %+ %%i, h
@@ -320,6 +327,7 @@ DECLARE_REG_TMP_SIZE 0,1,2,3,4,5,6,7,8,9
     %assign %%i 0
     %rep %0
         %xdefine %1q r %+ %%i %+ q
+        %xdefine %1p r %+ %%i %+ p
         %xdefine %1d r %+ %%i %+ d
         %xdefine %1w r %+ %%i %+ w
         %xdefine %1h r %+ %%i %+ h
@@ -1455,6 +1463,32 @@ FMA_INSTR pmadcswd, pmaddwd, paddd
 ; This lets us use tzcnt without bumping the yasm version requirement yet.
 %define tzcnt rep bsf
 
+; FIXME : integrate into DECLARE_REG macros
+
+%if ARCH_X86_64_X32
+%xdefine r0p r0d
+%xdefine r1p r1d
+%xdefine r2p r2d
+%xdefine r3p r3d
+%xdefine r4p r4d
+%xdefine r5p r5d
+%xdefine r6p r6d
+%xdefine r7p r7d
+%xdefine r8p r8d
+%xdefine r9p r9d
+%else
+%xdefine r0p r0
+%xdefine r1p r1
+%xdefine r2p r2
+%xdefine r3p r3
+%xdefine r4p r4
+%xdefine r5p r5
+%xdefine r6p r6
+%xdefine r7p r7
+%xdefine r8p r8
+%xdefine r9p r9
+%endif
+
 ; convert FMA4 to FMA3 if possible
 %macro FMA4_INSTR 4
     %macro %1 4-8 %1, %2, %3, %4
