diff -urp ffmpeg-2.8.6.orig/libswresample/x86/resample.asm ffmpeg-2.8.6/libswresample/x86/resample.asm
--- ffmpeg-2.8.6.orig/libswresample/x86/resample.asm	2016-10-22 10:36:36.378496945 -0700
+++ ffmpeg-2.8.6/libswresample/x86/resample.asm	2016-10-22 10:51:26.142471382 -0700
@@ -23,7 +23,7 @@
 %include "libavutil/x86/x86util.asm"
 
-%if ARCH_X86_64
+%if ARCH_X86_64_X64
 %define pointer resq
 %else
 %define pointer resd
 %endif
diff -urp ffmpeg-2.8.6.orig/libswresample/x86/resample.asm ffmpeg-2.8.6/libswresample/x86/resample.asm
--- ffmpeg-2.8.6.orig/libswresample/x86/resample.asm	2016-10-22 14:01:18.775157413 -0700
+++ ffmpeg-2.8.6/libswresample/x86/resample.asm	2016-10-22 14:52:29.991022993 -0700
@@ -65,8 +65,13 @@ cglobal resample_common_%1, 0, 15, 2, ct
                                       phase_mask, dst_end, filter_bank
 
     ; use red-zone for variable storage
-%define ctx_stackq            [rsp-0x8]
-%define src_stackq            [rsp-0x10]
+%ifdef __ILP32__
+%define ctx_stackp            [rsp-0x4]
+%define src_stackp            [rsp-0x8]
+%else
+%define ctx_stackp            [rsp-0x8]
+%define src_stackp            [rsp-0x10]
+%endif
 %if WIN64
 %define update_context_stackd r4m
 %else ; unix64
@@ -83,9 +90,9 @@ cglobal resample_common_%1, 0, 15, 2, ct
     mov                       indexd, [ctxq+ResampleContext.index]
     mov                        fracd, [ctxq+ResampleContext.frac]
     mov                dst_incr_modd, [ctxq+ResampleContext.dst_incr_mod]
-    mov                 filter_bankq, [ctxq+ResampleContext.filter_bank]
+    mov                 filter_bankp, [ctxq+ResampleContext.filter_bank]
     mov                    src_incrd, [ctxq+ResampleContext.src_incr]
-    mov                   ctx_stackq, ctxq
+    mov                   ctx_stackp, ctxp
     mov           min_filter_len_x4d, [ctxq+ResampleContext.filter_length]
     mov                dst_incr_divd, [ctxq+ResampleContext.dst_incr_div]
     shl           min_filter_len_x4d, %3
@@ -110,14 +117,14 @@ cglobal resample_common_%1, 0, 15, 2, ct
     neg           min_filter_len_x4q
     sub                 filter_bankq, min_filter_len_x4q
     sub                         srcq, min_filter_len_x4q
-    mov                   src_stackq, srcq
+    mov                   src_stackp, srcp
 %else ; x86-32
 cglobal resample_common_%1, 1, 7, 2, ctx, phase_shift, dst, frac, \
                                      index, min_filter_length_x4, filter_bank
 
     ; push temp variables to stack
-%define ctx_stackq            r0mp
-%define src_stackq            r2mp
+%define ctx_stackp            r0mp
+%define src_stackp            r2mp
 %define update_context_stackd r4m
 
     mov                         dstq, r1mp
@@ -134,7 +141,7 @@ cglobal resample_common_%1, 1, 7, 2, ctx
     shl        min_filter_length_x4d, %3
     mov                        fracd, [ctxq+ResampleContext.frac]
     neg        min_filter_length_x4q
-    mov                 filter_bankq, [ctxq+ResampleContext.filter_bank]
+    mov                 filter_bankp, [ctxq+ResampleContext.filter_bank]
     sub                         r2mp, min_filter_length_x4q
     sub                 filter_bankq, min_filter_length_x4q
     PUSH                              min_filter_length_x4q
@@ -253,10 +260,10 @@ cglobal resample_common_%1, 1, 7, 2, ctx
     ; strictly speaking, the function should always return the consumed
     ; number of bytes; however, we only use the value if update_context
     ; is true, so let's just leave it uninitialized otherwise
-    mov                         ctxq, ctx_stackq
+    mov                         ctxp, ctx_stackp
     movifnidn                    rax, srcq
     mov [ctxq+ResampleContext.frac ], fracd
-    sub                          rax, src_stackq
+    sub                          rax, src_stackp
     mov [ctxq+ResampleContext.index], indexd
     shr                          rax, %3
 
@@ -286,8 +293,13 @@ cglobal resample_linear_%1, 0, 15, 5, ct
 %endif
 
     ; use red-zone for variable storage
-%define ctx_stackq            [rsp-0x8]
-%define src_stackq            [rsp-0x10]
+%ifdef __ILP32__
+%define ctx_stackp            [rsp-0x4]
+%define src_stackp            [rsp-0x8]
+%else
+%define ctx_stackp            [rsp-0x8]
+%define src_stackp            [rsp-0x10]
+%endif
 %define phase_mask_stackd     [rsp-0x14]
 %if WIN64
 %define update_context_stackd r4m
@@ -307,7 +319,7 @@ cglobal resample_linear_%1, 0, 15, 5, ct
     mov                dst_incr_modd, [ctxq+ResampleContext.dst_incr_mod]
     mov                 filter_bankq, [ctxq+ResampleContext.filter_bank]
     mov                    src_incrd, [ctxq+ResampleContext.src_incr]
-    mov                   ctx_stackq, ctxq
+    mov                   ctx_stackp, ctxp
     mov            phase_mask_stackd, phase_maskd
     mov           min_filter_len_x4d, [ctxq+ResampleContext.filter_length]
 %ifidn %1, int16
@@ -340,14 +352,14 @@ cglobal resample_linear_%1, 0, 15, 5, ct
     neg           min_filter_len_x4q
     sub                 filter_bankq, min_filter_len_x4q
     sub                         srcq, min_filter_len_x4q
-    mov                   src_stackq, srcq
+    mov                   src_stackp, srcp
 %else ; x86-32
 cglobal resample_linear_%1, 1, 7, 5, ctx, min_filter_length_x4, filter2, \
                                      frac, index, dst, filter_bank
 
     ; push temp variables to stack
-%define ctx_stackq            r0mp
-%define src_stackq            r2mp
+%define ctx_stackp            r0mp
+%define src_stackp            r2mp
 %define update_context_stackd r4m
 
     mov                         dstq, r1mp
@@ -559,10 +571,10 @@ cglobal resample_linear_%1, 1, 7, 5, ctx
     ; strictly speaking, the function should always return the consumed
     ; number of bytes; however, we only use the value if update_context
     ; is true, so let's just leave it uninitialized otherwise
-    mov                         ctxq, ctx_stackq
+    mov                         ctxp, ctx_stackp
     movifnidn                    rax, srcq
     mov [ctxq+ResampleContext.frac ], fracd
-    sub                          rax, src_stackq
+    sub                          rax, src_stackp
     mov [ctxq+ResampleContext.index], indexd
     shr                          rax, %3
 
