diff -urp mono-4.4.0.orig/mono/mini/aot-compiler.c mono-4.4.0/mono/mini/aot-compiler.c
--- mono-4.4.0.orig/mono/mini/aot-compiler.c	2016-04-12 08:22:20.000000000 -0700
+++ mono-4.4.0/mono/mini/aot-compiler.c	2016-10-24 13:38:33.032585240 -0700
@@ -739,7 +739,11 @@ emit_code_bytes (MonoAotCompile *acfg, c
 #endif
 
 #ifdef TARGET_AMD64
-#define AOT_TARGET_STR "AMD64"
+#ifdef __mono_ilp32__
+#define AOT_TARGET_STR "AMD64 (mono ilp32)"
+#else
+#define AOT_TARGET_STR "AMD64 (!mono ilp32)"
+#endif
 #endif
 
 #ifdef TARGET_ARM
