diff -ur chromium-52.0.2743.116.orig/media/base/BUILD.gn chromium-52.0.2743.116/media/base/BUILD.gn
--- chromium-52.0.2743.116.orig/media/base/BUILD.gn	2016-08-03 12:02:19.000000000 -0700
+++ chromium-52.0.2743.116/media/base/BUILD.gn	2016-08-09 12:21:55.723305925 -0700
@@ -307,7 +307,7 @@
     defines += [ "DISABLE_USER_INPUT_MONITOR" ]
   }
 
-  if (current_cpu == "x86" || current_cpu == "x64") {
+  if (current_cpu == "x86" || current_cpu == "x64" || current_cpu == "x32") {
     sources += [
       "simd/convert_rgb_to_yuv_sse2.cc",
       "simd/convert_rgb_to_yuv_ssse3.cc",
@@ -487,7 +487,7 @@
     deps += [ "//ui/gl" ]
   }
 
-  if (current_cpu == "x86" || current_cpu == "x64") {
+  if (current_cpu == "x86" || current_cpu == "x64" || current_cpu == "x32") {
     sources += [ "simd/convert_rgb_to_yuv_unittest.cc" ]
   }
 }
@@ -523,7 +523,7 @@
   }
 }
 
-if (current_cpu == "x86" || current_cpu == "x64") {
+if (current_cpu == "x86" || current_cpu == "x64" || current_cpu == "x32") {
   import("//third_party/yasm/yasm_assemble.gni")
   yasm_assemble("media_yasm") {
     sources = [
@@ -562,7 +562,7 @@
 
     if (current_cpu == "x86") {
       yasm_flags += [ "-DARCH_X86_32" ]
-    } else if (current_cpu == "x64") {
+    } else if (current_cpu == "x64" || current_cpu == "x32") {
       yasm_flags += [ "-DARCH_X86_64" ]
       sources += [
         "simd/linear_scale_yuv_to_rgb_mmx_x64.asm",
@@ -578,7 +578,7 @@
     } else {
       if (is_posix) {
         yasm_flags += [ "-DELF" ]
-        if (current_cpu == "x64") {
+        if (current_cpu == "x64" || current_cpu == "x32") {
           # TODO(ajwong): Why isn't this true in mac?
           yasm_flags += [ "-DPIC" ]
         }
diff -ur chromium-52.0.2743.116.orig/media/media.gyp chromium-52.0.2743.116/media/media.gyp
--- chromium-52.0.2743.116.orig/media/media.gyp	2016-08-03 12:02:20.000000000 -0700
+++ chromium-52.0.2743.116/media/media.gyp	2016-08-09 12:24:02.150303902 -0700
@@ -1062,7 +1062,7 @@
             'filters/ffmpeg_h265_to_annex_b_bitstream_converter.h',
           ],
         }],
-        ['target_arch=="ia32" or target_arch=="x64"', {
+        ['target_arch=="ia32" or target_arch=="x64" or target_arch=="x32"', {
           'dependencies': [
             'media_asm',
           ],
@@ -1335,7 +1335,7 @@
             'filters/h264_bitstream_buffer_unittest.cc',
           ],
         }],
-        ['target_arch=="ia32" or target_arch=="x64"', {
+        ['target_arch=="ia32" or target_arch=="x64" or target_arch=="x32"', {
           'sources': [
             'base/simd/convert_rgb_to_yuv_unittest.cc',
           ],
@@ -1634,7 +1634,7 @@
     },
   ],
   'conditions': [
-    ['target_arch=="ia32" or target_arch=="x64"', {
+    ['target_arch=="ia32" or target_arch=="x64" or target_arch=="x32"', {
       'targets': [
        {
           'target_name': 'media_asm',
@@ -1655,7 +1655,7 @@
                 'yasm_flags': ['-DEXPORT_SYMBOLS'],
               },
             }],
-            ['target_arch=="x64"', {
+            ['target_arch=="x64" or target_arch=="x32"', {
               # Source files optimized for X64 systems.
               'sources': [
                 'base/simd/linear_scale_yuv_to_rgb_mmx_x64.asm',
