diff -ur chromium-51.0.2704.103.orig/v8/build/toolchain.gypi chromium-51.0.2704.103/v8/build/toolchain.gypi
--- chromium-51.0.2704.103.orig/v8/build/toolchain.gypi	2016-06-16 12:04:14.000000000 -0700
+++ chromium-51.0.2704.103/v8/build/toolchain.gypi	2016-06-30 13:32:05.395926146 -0700
@@ -38,6 +38,7 @@
     'ubsan%': 0,
     'ubsan_vptr%': 0,
     'v8_target_arch%': '<(target_arch)',
+    'v8_target_arch_abi%' : '<(target_arch_abi)',
     'v8_host_byteorder%': '<!(python -c "import sys; print sys.byteorder")',
     # Native Client builds currently use the V8 ARM JIT and
     # arm/simulator-arm.cc to defer the significant effort required
@@ -1042,7 +1043,7 @@
         ],
       }],
       ['(OS=="linux" or OS=="android") and \
-        (v8_target_arch=="x64" or v8_target_arch=="arm64" or \
+        ((v8_target_arch=="x64" and v8_target_arch_abi=="x64") or v8_target_arch=="arm64" or \
          v8_target_arch=="ppc64" or v8_target_arch=="s390x")', {
         'target_conditions': [
           ['_toolset=="host"', {
