diff -ur chromium-52.0.2743.116.orig/build/common.gypi chromium-52.0.2743.116/build/common.gypi
--- chromium-52.0.2743.116.orig/build/common.gypi	2016-08-08 12:53:55.989658279 -0700
+++ chromium-52.0.2743.116/build/common.gypi	2016-08-08 13:15:36.020637468 -0700
@@ -2444,7 +2444,7 @@
       # Do not disable seccomp_bpf anywhere without talking to
       # security@chromium.org!
       ['((OS=="linux" or OS=="android") and '
-           '(target_arch=="ia32" or target_arch=="x64" or '
+           '(target_arch=="ia32" or target_arch=="x64" or target_arch=="x32" or '
              'target_arch=="arm" or target_arch=="mipsel" or '
              'target_arch=="arm64"))', {
          'use_seccomp_bpf%': 1,
