diff -urp chromium-52.0.2743.116.orig/v8/BUILD.gn chromium-52.0.2743.116/v8/BUILD.gn
--- chromium-52.0.2743.116.orig/v8/BUILD.gn	2016-08-03 12:04:09.000000000 -0700
+++ chromium-52.0.2743.116/v8/BUILD.gn	2016-08-17 12:57:09.894207338 -0700
@@ -291,7 +291,7 @@ config("toolchain") {
   if (v8_target_arch == "x86") {
     defines += [ "V8_TARGET_ARCH_IA32" ]
   }
-  if (v8_target_arch == "x64") {
+  if (v8_target_arch == "x64" || v8_target_arch == "x32") {
     defines += [ "V8_TARGET_ARCH_X64" ]
   }
 
@@ -1557,7 +1557,7 @@ v8_source_set("v8_base") {
       "src/regexp/ia32/regexp-macro-assembler-ia32.cc",
       "src/regexp/ia32/regexp-macro-assembler-ia32.h",
     ]
-  } else if (v8_target_arch == "x64") {
+  } else if (v8_target_arch == "x64" || v8_target_arch == "x32") {
     sources += [
       "src/compiler/x64/code-generator-x64.cc",
       "src/compiler/x64/instruction-codes-x64.h",
diff -urp chromium-52.0.2743.116.orig/v8/src/x64/disasm-x64.cc chromium-52.0.2743.116/v8/src/x64/disasm-x64.cc
--- chromium-52.0.2743.116.orig/v8/src/x64/disasm-x64.cc	2016-08-03 12:04:09.000000000 -0700
+++ chromium-52.0.2743.116/v8/src/x64/disasm-x64.cc	2016-08-17 12:18:59.112244009 -0700
@@ -1864,6 +1864,7 @@ int DisassemblerX64::TwoByteOpcodeInstru
     // SETcc: Set byte on condition. Needs pointer to beginning of instruction.
     current = data + SetCC(data);
 
+#if V8_HOST_ARCH_64_BIT
   } else if (opcode == 0xAB || opcode == 0xA5 || opcode == 0xAD) {
     // SHLD, SHRD (double-precision shift), BTS (bit set).
     AppendToBuffer("%s ", mnemonic);
@@ -1875,6 +1876,30 @@ int DisassemblerX64::TwoByteOpcodeInstru
     } else {
       AppendToBuffer(",%s,cl", NameOfCPURegister(regop));
     }
+#else //V8_HOST_ARCH_32_BIT
+    } else if (opcode == 0xA4 || opcode == 0xAC) {
+      // shld, shrd
+      //current += 2;
+      AppendToBuffer("%s ", mnemonic);
+      int mod, regop, rm;
+      get_modrm(*current, &mod, &regop, &rm);
+      int8_t imm8 = static_cast<int8_t>(current[1]);
+      current += 2;
+      AppendToBuffer("%s,%s,%d", NameOfCPURegister(rm),
+                     NameOfCPURegister(regop), static_cast<int>(imm8));
+    } else if (opcode == 0xAB || opcode == 0xA5 || opcode == 0xAD) {
+      // shrd_cl, shld_cl, bts
+      //current += 2;
+      AppendToBuffer("%s ", mnemonic);
+      int mod, regop, rm;
+      get_modrm(*current, &mod, &regop, &rm);
+      current += PrintRightOperand(current);
+      if (opcode == 0xAB) {
+        AppendToBuffer(",%s", NameOfCPURegister(regop));
+      } else {
+        AppendToBuffer(",%s,cl", NameOfCPURegister(regop));
+    }
+#endif //V8_HOST_ARCH_32_BIT
   } else if (opcode == 0xB8 || opcode == 0xBC || opcode == 0xBD) {
     // POPCNT, CTZ, CLZ.
     AppendToBuffer("%s%c ", mnemonic, operand_size_code());
@@ -1918,12 +1943,23 @@ const char* DisassemblerX64::TwoByteMnem
       return (group_1_prefix_ == 0xF2) ? "maxsd" : "maxss";
     case 0xA2:
       return "cpuid";
+#if V8_HOST_ARCH_64_BIT
+    case 0xA5:
+      return "shld";
+    case 0xAD:
+      return "shrd";
+#else //V8_HOST_ARCH_32_BIT
+    case 0xA4:
+      return "shld";
     case 0xA5:
       return "shld";
-    case 0xAB:
-      return "bts";
     case 0xAD:
       return "shrd";
+    case 0xAC:
+      return "shrd";  // 3-operand version.
+#endif //V8_HOST_ARCH_32_BIT
+    case 0xAB:
+      return "bts";
     case 0xAF:
       return "imul";
     case 0xB6:
