diff -urp -x '*.rej' -x '*.orig' chromium-54.0.2824.0.orig/v8/src/builtins/x64/builtins-x64.cc chromium-54.0.2824.0/v8/src/builtins/x64/builtins-x64.cc
--- chromium-54.0.2824.0.orig/v8/src/builtins/x64/builtins-x64.cc	2016-08-09 22:46:55.000000000 -0700
+++ chromium-54.0.2824.0/v8/src/builtins/x64/builtins-x64.cc	2016-08-17 18:09:12.814907622 -0700
@@ -538,8 +538,8 @@ void Builtins::Generate_ResumeGeneratorT
     // the generator was suspended.
     FrameScope scope(masm, StackFrame::MANUAL);
     __ PushReturnAddressFrom(rax);  // Return address.
-    __ Push(rbp);                   // Caller's frame pointer.
-    __ Move(rbp, rsp);
+    __ pushq(rbp);                   // Caller's frame pointer.
+    __ movp(rbp, rsp);
     __ Push(rsi);  // Callee's context.
     __ Push(rdi);  // Callee's JS Function.
 
diff -urp -x '*.rej' -x '*.orig' chromium-54.0.2824.0.orig/v8/src/crankshaft/x64/lithium-codegen-x64.cc chromium-54.0.2824.0/v8/src/crankshaft/x64/lithium-codegen-x64.cc
--- chromium-54.0.2824.0.orig/v8/src/crankshaft/x64/lithium-codegen-x64.cc	2016-08-09 22:46:55.000000000 -0700
+++ chromium-54.0.2824.0/v8/src/crankshaft/x64/lithium-codegen-x64.cc	2016-08-17 18:10:21.797906517 -0700
@@ -2940,8 +2940,16 @@ Operand LCodeGen::BuildFastArrayOperand(
     return Operand(elements_pointer_reg,
                    (constant_value << shift_size) + offset);
   } else {
+#if V8_TARGET_ARCH_64_BIT
     // Guaranteed by ArrayInstructionInterface::KeyedAccessIndexRequirement().
     DCHECK(key_representation.IsInteger32());
+#else //V8_TARGET_ARCH_32_BIT
+    // Take the tag bit into account while computing the shift size.
+    //if (key_representation.IsSmi() && (shift_size >= 1)) {
+    //  DCHECK(SmiValuesAre31Bits());
+    //  shift_size -= kSmiTagSize;
+    //}
+#endif //V8_TARGET_ARCH_32_BIT
 
     ScaleFactor scale_factor = static_cast<ScaleFactor>(shift_size);
     return Operand(elements_pointer_reg,
