--- chromium-53.0.2785.143/third_party/WebKit/Source/BUILD.gn.orig	2016-11-16 02:04:51.007400440 -0800
+++ chromium-53.0.2785.143/third_party/WebKit/Source/BUILD.gn	2016-11-16 02:08:25.930730944 -0800
@@ -70,6 +70,12 @@ config("config") {
     }
   }
 
+  if (target_arch=="x32") {
+    cflags += [
+      "-Wno-array-bounds",
+    ]
+  }
+
   if (is_clang && blink_gc_plugin && clang_use_chrome_plugins) {
     # On Windows, the plugin is built directly into clang, so there's
     # no need to load it dynamically.

