diff -ur chromium-52.0.2743.116.orig/third_party/libjpeg_turbo/libjpeg.gyp chromium-52.0.2743.116/third_party/libjpeg_turbo/libjpeg.gyp
--- chromium-52.0.2743.116.orig/third_party/libjpeg_turbo/libjpeg.gyp	2016-08-08 23:24:10.271052947 -0700
+++ chromium-52.0.2743.116/third_party/libjpeg_turbo/libjpeg.gyp	2016-08-09 00:17:01.870002176 -0700
@@ -167,6 +167,11 @@
             'simd/jquanti-sse2-64.asm',
           ],
         }],
+        [ 'target_arch=="x32"', {
+          'sources': [
+            'jsimd_none.c',
+          ],
+        }],
         # MemorySanitizer doesn't support assembly code, so keep it disabled in
         # MSan builds for now.
         [ 'msan==1', {
