diff -ur chromium-51.0.2704.103.orig/build/common.gypi chromium-51.0.2704.103/build/common.gypi
--- chromium-51.0.2704.103.orig/build/common.gypi	2016-06-29 17:35:03.739178898 -0700
+++ chromium-51.0.2704.103/build/common.gypi	2016-06-29 17:38:54.824181302 -0700
@@ -4600,6 +4600,10 @@
           ['use_allocator!="tcmalloc"', {
             'defines': ['NO_TCMALLOC'],
           }],
+          ['os_posix==1', {
+            'defines': ['_GLIBCXX_USE_C99_CHECK', '__gnuc_va_list=__isoc_va_list', '_GLIBCXX_HAVE_BROKEN_VSWPRINTF'],
+            'cflags': ['-fpermissive'],
+          }],
           ['linux_use_gold_flags==1', {
             # Newer gccs and clangs support -fuse-ld, use the flag to force gold
             # selection.
