diff -urp chromium-53.0.2785.143.orig/v8/include/v8-platform.h chromium-53.0.2785.143/v8/include/v8-platform.h
--- chromium-53.0.2785.143.orig/v8/include/v8-platform.h	2016-10-05 17:22:15.769138502 -0700
+++ chromium-53.0.2785.143/v8/include/v8-platform.h	2016-10-05 19:02:25.731634502 -0700
@@ -20,6 +20,18 @@ class Task {
 };
 
 /**
+* An IdleTask represents a unit of work to be performed in idle time.
+* The Run method is invoked with an argument that specifies the deadline in
+* seconds returned by MonotonicallyIncreasingTime().
+* The idle task is expected to complete by this deadline.
+*/
+class IdleTask {
+ public:
+  virtual ~IdleTask() {}
+  virtual void Run(double deadline_in_seconds) = 0;
+};
+
+/**
  * V8 Platform abstraction layer.
  *
  * The embedder has to provide an implementation of this interface before
