diff -ur chromium-52.0.2743.116.orig/v8/BUILD.gn chromium-52.0.2743.116/v8/BUILD.gn
--- chromium-52.0.2743.116.orig/v8/BUILD.gn	2016-10-04 22:25:35.720432687 -0700
+++ chromium-52.0.2743.116/v8/BUILD.gn	2016-10-04 23:09:55.099943128 -0700
@@ -633,13 +633,6 @@
     "src/jsregexp-inl.h",
     "src/jsregexp.cc",
     "src/jsregexp.h",
-    # TODO(jochen): move libplatform/ files to their own target.
-    "src/libplatform/default-platform.cc",
-    "src/libplatform/default-platform.h",
-    "src/libplatform/task-queue.cc",
-    "src/libplatform/task-queue.h",
-    "src/libplatform/worker-thread.cc",
-    "src/libplatform/worker-thread.h",
     "src/list-inl.h",
     "src/list.h",
     "src/lithium-allocator-inl.h",
@@ -1069,6 +1062,25 @@
   configs += [ ":internal_config", ":features", ":toolchain" ]
 }
 
+v8_source_set("v8_libplatform") {
+  sources = [
+    "src/libplatform/default-platform.cc",
+    "src/libplatform/default-platform.h",
+    "src/libplatform/task-queue.cc",
+    "src/libplatform/task-queue.h",
+    "src/libplatform/worker-thread.cc",
+    "src/libplatform/worker-thread.h",
+  ]
+
+  configs = [ ":internal_config_base" ]
+
+  public_configs = [ ":libplatform_config" ]
+
+  deps = [
+    ":v8_libbase",
+  ]
+}
+
 ###############################################################################
 # Executables
 #
