diff -ur chromium-52.0.2743.116.orig/v8/samples/samples.gyp chromium-52.0.2743.116/v8/samples/samples.gyp
--- chromium-52.0.2743.116.orig/v8/samples/samples.gyp	2016-10-04 21:19:41.953365981 -0700
+++ chromium-52.0.2743.116/v8/samples/samples.gyp	2016-10-04 21:19:14.241293774 -0700
@@ -55,7 +55,7 @@
   },
   'targets': [
     {
-      'target_name': 'shell',
+      'target_name': 'v8_shell',
       'sources': [
         'v8_shell.cc',
       ],
diff -ur chromium-52.0.2743.116.orig/v8/tools/gyp/v8.gyp chromium-52.0.2743.116/v8/tools/gyp/v8.gyp
--- chromium-52.0.2743.116.orig/v8/tools/gyp/v8.gyp	2016-10-04 21:03:05.887827248 -0700
+++ chromium-52.0.2743.116/v8/tools/gyp/v8.gyp	2016-10-04 21:18:28.499523981 -0700
@@ -529,13 +529,6 @@
         '../../src/jsregexp-inl.h',
         '../../src/jsregexp.cc',
         '../../src/jsregexp.h',
-        # TODO(jochen): move libplatform/ files to their own target.
-        '../../src/libplatform/default-platform.cc',
-        '../../src/libplatform/default-platform.h',
-        '../../src/libplatform/task-queue.cc',
-        '../../src/libplatform/task-queue.h',
-        '../../src/libplatform/worker-thread.cc',
-        '../../src/libplatform/worker-thread.h',
         '../../src/list-inl.h',
         '../../src/list.h',
         '../../src/lithium-allocator-inl.h',
@@ -1182,6 +1175,40 @@
       ],
     },
     {
+      'target_name': 'v8_libplatform',
+      'type': 'static_library',
+      'variables': {
+        'optimize': 'max',
+      },
+      'dependencies': [
+        'v8_libbase',
+      ],
+      'include_dirs+': [
+        '../..',
+        '../../include',
+      ],
+      'sources': [
+        '../../src/libplatform/default-platform.cc',
+        '../../src/libplatform/default-platform.h',
+        '../../src/libplatform/task-queue.cc',
+        '../../src/libplatform/task-queue.h',
+        '../../src/libplatform/worker-thread.cc',
+        '../../src/libplatform/worker-thread.h',
+      ],
+      'conditions': [
+        ['want_separate_host_toolset==1', {
+          'toolsets': ['host', 'target'],
+        }, {
+          'toolsets': ['target'],
+        }],
+      ],
+      'direct_dependent_settings': {
+        'include_dirs': [
+          '../../include',
+        ],
+      },
+    },
+    {
       'target_name': 'natives_blob',
       'type': 'none',
       'conditions': [
