diff -ur chromium-51.0.2704.103.orig/build/config/BUILDCONFIG.gn chromium-51.0.2704.103/build/config/BUILDCONFIG.gn
--- chromium-51.0.2704.103.orig/build/config/BUILDCONFIG.gn	2016-06-16 12:02:01.000000000 -0700
+++ chromium-51.0.2704.103/build/config/BUILDCONFIG.gn	2016-06-30 12:53:00.266901747 -0700
@@ -53,6 +53,9 @@
 if (current_cpu == "") {
   current_cpu = target_cpu
 }
+if (current_cpu_abi == "") {
+  current_cpu_abi = target_cpu_abi
+}
 if (current_os == "") {
   current_os = target_os
 }
diff -ur chromium-51.0.2704.103.orig/build/config/compiler/BUILD.gn chromium-51.0.2704.103/build/config/compiler/BUILD.gn
--- chromium-51.0.2704.103.orig/build/config/compiler/BUILD.gn	2016-06-16 12:02:01.000000000 -0700
+++ chromium-51.0.2704.103/build/config/compiler/BUILD.gn	2016-06-30 12:51:53.437901052 -0700
@@ -451,11 +451,20 @@
     # CPU architecture. We may or may not be doing a cross compile now, so for
     # simplicity we always explicitly set the architecture.
     if (current_cpu == "x64") {
-      cflags += [
-        "-m64",
-        "-march=x86-64",
-      ]
-      ldflags += [ "-m64" ]
+      if (current_cpu_abi == "x64") {
+        cflags += [
+          "-m64",
+          "-march=x86-64",
+        ]
+        ldflags += [ "-m64" ]
+      }
+      if (current_cpu_abi == "x32") {
+        cflags += [
+          "-mx32",
+          "-march=x86-64",
+        ]
+        ldflags += [ "-mx32" ]
+      }
     } else if (current_cpu == "x86") {
       cflags += [ "-m32" ]
       ldflags += [ "-m32" ]
