diff -ur chromium-54.0.2824.0.orig/v8/gypfiles/toolchain.gypi chromium-54.0.2824.0/v8/gypfiles/toolchain.gypi
--- chromium-54.0.2824.0.orig/v8/gypfiles/toolchain.gypi	2016-08-09 22:46:55.000000000 -0700
+++ chromium-54.0.2824.0/v8/gypfiles/toolchain.gypi	2016-08-17 16:42:23.388991014 -0700
@@ -40,6 +40,7 @@
     'has_valgrind%': 0,
     'coverage%': 0,
     'v8_target_arch%': '<(target_arch)',
+    'v8_target_arch_abi%': '<(target_arch_abi)',
     'v8_host_byteorder%': '<!(python -c "import sys; print sys.byteorder")',
     'force_dynamic_crt%': 0,
 
@@ -1064,7 +1065,7 @@
         ],
       }],
       ['(OS=="linux" or OS=="android") and \
-        (v8_target_arch=="x64" or v8_target_arch=="arm64" or \
+        ((v8_target_arch=="x64" and v8_target_arch_abi=="x64") or v8_target_arch=="arm64" or \
          v8_target_arch=="ppc64" or v8_target_arch=="s390x")', {
         'target_conditions': [
           ['_toolset=="host"', {
Only in chromium-54.0.2824.0.orig/v8: v8
