diff -urp chromium-54.0.2824.0.orig/ui/gfx/gfx.gyp chromium-54.0.2824.0/ui/gfx/gfx.gyp
--- chromium-54.0.2824.0.orig/ui/gfx/gfx.gyp	2016-08-09 22:45:34.000000000 -0700
+++ chromium-54.0.2824.0/ui/gfx/gfx.gyp	2016-08-17 19:07:56.833851209 -0700
@@ -82,7 +82,7 @@
         '<(DEPTH)/third_party/icu/icu.gyp:icui18n',
         '<(DEPTH)/third_party/icu/icu.gyp:icuuc',
         '<(DEPTH)/third_party/libpng/libpng.gyp:libpng',
-	'<(DEPTH)//third_party/qcms/qcms.gyp:qcms",
+	'<(DEPTH)/third_party/qcms/qcms.gyp:qcms',
         '<(DEPTH)/third_party/zlib/zlib.gyp:zlib',
         'gfx_geometry',
         'gfx_range',
