diff -ur chromium-52.0.2743.116.orig/v8/src/v8.gyp chromium-52.0.2743.116/v8/src/v8.gyp
--- chromium-52.0.2743.116.orig/v8/src/v8.gyp	2016-08-03 12:04:09.000000000 -0700
+++ chromium-52.0.2743.116/v8/src/v8.gyp	2016-08-04 19:09:00.436830399 -0700
@@ -1484,7 +1484,7 @@
             'regexp/x64/regexp-macro-assembler-x64.h',
           ],
         }],
-        ['v8_target_arch=="x64"', {
+        ['v8_target_arch=="x64" or v8_target_arch=="x32"', {
           'sources': [
             'compiler/x64/code-generator-x64.cc',
             'compiler/x64/instruction-codes-x64.h',
