diff -ur chromium-52.0.2743.116.orig/third_party/libjpeg_turbo/libjpeg.gyp chromium-52.0.2743.116/third_party/libjpeg_turbo/libjpeg.gyp
--- chromium-52.0.2743.116.orig/third_party/libjpeg_turbo/libjpeg.gyp	2016-08-08 13:20:39.865632604 -0700
+++ chromium-52.0.2743.116/third_party/libjpeg_turbo/libjpeg.gyp	2016-08-08 22:26:50.532108010 -0700
@@ -23,9 +23,18 @@
       'include_dirs': [
         '.',
       ],
-      'defines': [
-        'WITH_SIMD',
-        'NO_GETENV',
+      'conditions': [
+        [ 'target_arch=="x32"', {
+          'defines': [
+            'NO_GETENV',
+          ],
+        }],
+        [ 'target_arch!="x32"', {
+          'defines': [
+            'WITH_SIMD',
+            'NO_GETENV',
+          ],
+        }],
       ],
       'sources': [
         'jcapimin.c',
