diff -urp chromium-52.0.2743.116.orig/v8/BUILD.gn chromium-52.0.2743.116/v8/BUILD.gn
--- chromium-52.0.2743.116.orig/v8/BUILD.gn	2016-10-04 23:58:51.881209331 -0700
+++ chromium-52.0.2743.116/v8/BUILD.gn	2016-10-05 00:14:50.249190357 -0700
@@ -1100,6 +1100,7 @@ if (current_toolchain == host_toolchain)
 
     deps = [
       ":v8_base",
+      ":v8_libplatform",
       ":v8_nosnapshot",
     ]
 
diff -urp chromium-52.0.2743.116.orig/v8/tools/gyp/v8.gyp chromium-52.0.2743.116/v8/tools/gyp/v8.gyp
--- chromium-52.0.2743.116.orig/v8/tools/gyp/v8.gyp	2016-10-04 23:58:51.758204584 -0700
+++ chromium-52.0.2743.116/v8/tools/gyp/v8.gyp	2016-10-05 00:14:47.995103388 -0700
@@ -1382,7 +1382,7 @@
     {
       'target_name': 'mksnapshot',
       'type': 'executable',
-      'dependencies': ['v8_base', 'v8_nosnapshot'],
+      'dependencies': ['v8_base', 'v8_nosnapshot', 'v8_libplatform'],
       'include_dirs+': [
         '../..',
       ],
