diff -ur chromium-54.0.2824.0.orig/third_party/libvpx/BUILD.gn chromium-54.0.2824.0/third_party/libvpx/BUILD.gn
--- chromium-54.0.2824.0.orig/third_party/libvpx/BUILD.gn	2016-08-17 17:30:32.442944766 -0700
+++ chromium-54.0.2824.0/third_party/libvpx/BUILD.gn	2016-08-17 17:48:42.400927318 -0700
@@ -12,7 +12,7 @@
 if (current_cpu == "x86") {
   cpu_arch_full = "ia32"
 } else if (current_cpu == "x64" || current_cpu == "x32") {
-  if (is_msan) {
+  if (is_msan || current_cpu == "x32") {
     cpu_arch_full = "generic"
   } else {
     cpu_arch_full = "x64"
@@ -275,7 +275,7 @@
   } else if (current_cpu == "x86") {
     sources = libvpx_srcs_x86
   } else if (current_cpu == "x64" || current_cpu == "x32") {
-    if (is_msan) {
+    if (is_msan || current_cpu == "x32") {
       sources = libvpx_srcs_generic
     } else {
       sources = libvpx_srcs_x86_64
@@ -299,7 +299,7 @@
   configs += [ "//build/config/compiler:no_chromium_code" ]
   configs += [ ":libvpx_warnings" ]
   deps = []
-  if (current_cpu == "x86" || current_cpu == "x32"  || (current_cpu == "x64" && !is_msan)) {
+  if (current_cpu == "x86" || (current_cpu == "x64" && !is_msan)) {
     deps += [
       ":libvpx_intrinsics_avx",
       ":libvpx_intrinsics_avx2",
Only in chromium-54.0.2824.0/third_party/libvpx: BUILD.gn.orig
Only in chromium-54.0.2824.0/third_party/libvpx: BUILD.gn.rej
diff -ur chromium-54.0.2824.0.orig/third_party/libvpx/libvpx.gyp chromium-54.0.2824.0/third_party/libvpx/libvpx.gyp
--- chromium-54.0.2824.0.orig/third_party/libvpx/libvpx.gyp	2016-08-17 17:30:32.561944764 -0700
+++ chromium-54.0.2824.0/third_party/libvpx/libvpx.gyp	2016-08-17 17:35:27.821940038 -0700
@@ -35,7 +35,7 @@
             'target_arch_full': 'arm64',
           }],
           ['target_arch=="x32"', {
-            'target_arch_full': 'x64',
+            'target_arch_full': 'generic',
           }],
         ],
       }],
@@ -92,7 +92,7 @@
     ['target_arch=="ia32"', {
       'includes': ['libvpx_srcs_x86_intrinsics.gypi', ],
     }],
-    ['(target_arch=="x64" or target_arch=="x32") and msan==0', {
+    ['(target_arch=="x64") and msan==0', {
       'includes': ['libvpx_srcs_x86_64_intrinsics.gypi', ],
     }],
     [ '(target_arch=="arm" or target_arch=="armv7") and arm_neon==0 and OS=="android"', {
@@ -171,7 +171,7 @@
             }],
             ['target_arch=="x64" or target_arch=="x32"', {
               'conditions': [
-                ['msan==1', {
+                ['msan==1 or target_arch=="x32"', {
                   'includes': [ 'libvpx_srcs_generic.gypi', ],
                 }, {
                   'includes': [
