diff -ur musl-1.1.14.orig/src/network/sendmsg.c musl-1.1.14/src/network/sendmsg.c
--- musl-1.1.14.orig/src/network/sendmsg.c	2016-02-21 21:07:05.000000000 -0800
+++ musl-1.1.14/src/network/sendmsg.c	2016-09-01 13:47:48.610306708 -0700
@@ -7,7 +7,7 @@
 
 ssize_t sendmsg(int fd, const struct msghdr *msg, int flags)
 {
-#if LONG_MAX > INT_MAX
+#if LONG_MAX > INT_MAX || (defined(__x86_64__) && defined(__ILP32__))
 	struct msghdr h;
 	struct cmsghdr chbuf[1024/sizeof(struct cmsghdr)+1], *c;
 	if (msg) {
