diff -urp busybox-1.20.2.orig/networking/ifplugd.c busybox-1.20.2/networking/ifplugd.c
--- busybox-1.20.2.orig/networking/ifplugd.c	2012-07-02 07:08:25.000000000 -0700
+++ busybox-1.20.2/networking/ifplugd.c	2017-09-08 18:49:16.205469206 -0700
@@ -38,7 +38,9 @@
 #include <linux/mii.h>
 #include <linux/ethtool.h>
 #ifdef HAVE_NET_ETHERNET_H
+# if defined(__UCLIBC__) || defined(__GLIBC__)
 # include <net/ethernet.h>
+# endif
 #endif
 #include <linux/netlink.h>
 #include <linux/rtnetlink.h>
