diff -urp genkernel-3.4.52.4.orig/gen_initramfs.sh genkernel-3.4.52.4/gen_initramfs.sh
--- genkernel-3.4.52.4.orig/gen_initramfs.sh	2016-01-05 15:55:05.000000000 -0800
+++ genkernel-3.4.52.4/gen_initramfs.sh	2017-09-08 20:35:34.856022527 -0700
@@ -506,11 +506,13 @@ append_linker() {
 
 	mkdir -p "${TEMP}/initramfs-linker-temp/etc/ld.so.conf.d"
 
-	cp "/etc/ld.so."{cache,conf} "${TEMP}/initramfs-linker-temp/etc/" 2> /dev/null \
-		|| gen_die "Could not copy ld.so.{cache,conf}"
+	cp "/etc/ld.so.cache" "${TEMP}/initramfs-linker-temp/etc/" 2> /dev/null #\
+		#|| gen_die "Could not copy ld.so.cache"
+	cp "/etc/ld.so.conf" "${TEMP}/initramfs-linker-temp/etc/" 2> /dev/null \
+		|| gen_die "Could not copy ld.so.conf"
 
-	cp -r "/etc/ld.so.conf.d" "${TEMP}/initramfs-linker-temp/etc/" 2> /dev/null \
-		|| gen_die "Could not copy ld.so.conf.d"
+	cp -r "/etc/ld.so.conf.d" "${TEMP}/initramfs-linker-temp/etc/" 2> /dev/null #\
+	#	|| gen_die "Could not copy ld.so.conf.d"
 
 	cd "${TEMP}/initramfs-linker-temp/"
 	log_future_cpio_content
