diff -urp screen-4.8.0.orig/configure.ac screen-4.8.0/configure.ac
--- screen-4.8.0.orig/configure.ac	2020-02-05 12:09:38.000000000 -0800
+++ screen-4.8.0/configure.ac	2020-11-29 17:13:56.672160972 -0800
@@ -179,26 +179,6 @@ AC_EGREP_CPP(YES_IS_DEFINED,
 #endif
 ], LIBS="$LIBS -lsocket -linet";seqptx=1)
 
-AC_CHECKING(SVR4)
-AC_EGREP_CPP(yes,
-[main () {
-#if defined(SVR4) || defined(__SVR4)
-  yes;
-#endif
-], AC_NOTE(- you have a SVR4 system) AC_DEFINE(SVR4) svr4=1)
-if test -n "$svr4" ; then
-oldlibs="$LIBS"
-LIBS="$LIBS -lelf"
-AC_CHECKING(SVR4)
-AC_TRY_LINK([#include <utmpx.h>
-],,
-[AC_CHECK_HEADER(dwarf.h, AC_DEFINE(BUGGYGETLOGIN),
-[AC_CHECK_HEADER(elf.h, AC_DEFINE(BUGGYGETLOGIN))])]
-,LIBS="$oldlibs")
-fi
-
-AC_CHECK_HEADERS([stropts.h string.h strings.h])
-
 AC_CHECKING(for Solaris 2.x)
 AC_EGREP_CPP(YES_IS_DEFINED,
 [#if defined(SVR4) && defined(sun)
diff -urp screen-4.8.0.orig/utmp.c screen-4.8.0/utmp.c
--- screen-4.8.0.orig/utmp.c	2020-02-05 12:09:38.000000000 -0800
+++ screen-4.8.0/utmp.c	2020-11-29 17:14:15.983904933 -0800
@@ -33,6 +33,7 @@
 #include "config.h"
 #include "screen.h"
 #include "extern.h"
+#include "os.h"
 
 #ifdef HAVE_UTEMPTER
 #include <utempter.h>
