diff -ur at-spi2-atk-2.18.1.orig/Makefile.in at-spi2-atk-2.18.1/Makefile.in
--- at-spi2-atk-2.18.1.orig/Makefile.in	2016-06-27 03:25:09.701087103 -0700
+++ at-spi2-atk-2.18.1/Makefile.in	2016-06-27 03:26:17.859087812 -0700
@@ -396,19 +396,6 @@
 .SUFFIXES:
 am--refresh: Makefile
 	@:
-$(srcdir)/Makefile.in:  $(srcdir)/Makefile.am  $(am__configure_deps)
-	@for dep in $?; do \
-	  case '$(am__configure_deps)' in \
-	    *$$dep*) \
-	      echo ' cd $(srcdir) && $(AUTOMAKE) --foreign'; \
-	      $(am__cd) $(srcdir) && $(AUTOMAKE) --foreign \
-		&& exit 0; \
-	      exit 1;; \
-	  esac; \
-	done; \
-	echo ' cd $(top_srcdir) && $(AUTOMAKE) --foreign Makefile'; \
-	$(am__cd) $(top_srcdir) && \
-	  $(AUTOMAKE) --foreign Makefile
 Makefile: $(srcdir)/Makefile.in $(top_builddir)/config.status
 	@case '$?' in \
 	  *config.status*) \
