diff -ur firefox-45.2.0esr.orig/configure.in firefox-45.2.0esr/configure.in
--- firefox-45.2.0esr.orig/configure.in	2016-07-30 05:10:37.692254104 -0700
+++ firefox-45.2.0esr/configure.in	2016-07-30 05:23:25.545279406 -0700
@@ -6164,9 +6164,16 @@
     fi
   ;;
   *:x86_64)
-    if $CC -E -dM -</dev/null | grep -q __ELF__; then
-      LIBJPEG_TURBO_ASFLAGS="-f elf64 -rnasm -pnasm -D__x86_64__ -DPIC -DELF"
-    fi
+    case "$target" in
+    *-gnux32 | *-muslx32)
+      LIBJPEG_TURBO_ASFLAGS="-f elfx32 -rnasm -pnasm -D__x86_64__ -DPIC -DELF"
+      ;;
+    *)
+      if $CC -E -dM -</dev/null | grep -q __ELF__; then
+        LIBJPEG_TURBO_ASFLAGS="-f elf64 -rnasm -pnasm -D__x86_64__ -DPIC -DELF"
+      fi
+      ;;
+    esac
   ;;
   esac
 fi
