--- a/configure.in.orig	2018-04-02 02:35:19.412754729 +0000
+++ b/configure.in	2018-04-02 02:36:33.297755498 +0000
@@ -5845,10 +5845,17 @@ if test -n "$MOZ_LIBJPEG_TURBO"; then
     fi
   ;;
   *:x86_64)
-    if $CC -E -dM -</dev/null | grep -q __ELF__; then
-      LIBJPEG_TURBO_ASFLAGS="-f elf64 -rnasm -pnasm -D__x86_64__ -DPIC -DELF"
-      LIBJPEG_TURBO_X64_ASM=1
-    fi
+    case "$target" in
+    *-gnux32 | *-muslx32)
+      # yasm doesn't support elfx32.
+      ;;
+    *)
+      if $CC -E -dM -</dev/null | grep -q __ELF__; then
+        LIBJPEG_TURBO_ASFLAGS="-f elf64 -rnasm -pnasm -D__x86_64__ -DPIC -DELF"
+        LIBJPEG_TURBO_X64_ASM=1
+      fi
+      ;;
+    esac
   ;;
   esac
 
