--- nano-2.7.5.ebuild.orig	2020-11-28 22:26:34.000000000 -0800
+++ nano-2.7.5.ebuild	2020-11-28 22:27:15.495525147 -0800
@@ -10,7 +10,7 @@ if [[ ${PV} == "9999" ]] ; then
 else
 	MY_P=${PN}-${PV/_}
 	SRC_URI="https://www.nano-editor.org/dist/v${PV:0:3}/${MY_P}.tar.gz"
-	KEYWORDS="alpha amd64 arm arm64 hppa ia64 ~m68k ~mips ppc ppc64 ~s390 ~sh sparc x86 ~ppc-aix ~amd64-fbsd ~sparc-fbsd ~x86-fbsd ~amd64-linux ~arm-linux ~x86-linux ~ppc-macos ~x64-macos ~x86-macos ~m68k-mint ~sparc-solaris ~sparc64-solaris ~x64-solaris ~x86-solaris"
+	KEYWORDS="~alpha amd64 ~arm ~arm64 hppa ~ia64 ~m68k ~mips ~ppc ~ppc64 ~s390 ~sh ~sparc ~x86 ~ppc-aix ~amd64-fbsd ~sparc-fbsd ~x86-fbsd ~amd64-linux ~arm-linux ~x86-linux ~ppc-macos ~x64-macos ~x86-macos ~m68k-mint ~sparc-solaris ~sparc64-solaris ~x64-solaris ~x86-solaris"
 fi
 
 DESCRIPTION="GNU GPL'd Pico clone with more functionality"
@@ -29,9 +29,13 @@ RDEPEND="!static? ( ${LIB_DEPEND//\[stat
 DEPEND="${RDEPEND}
 	nls? ( sys-devel/gettext )
 	virtual/pkgconfig
-	static? ( ${LIB_DEPEND} )"
+	static? ( ${LIB_DEPEND} )
+	app-shells/bash" #line added by muslx32 overlay.  bash needs to be reemerged before the nano package.
 
 src_prepare() {
+	if [[ "${CHOST}" =~ "muslx32" ]] ; then
+		sed -i "s|limits.h regex.h sys|limits.h sys|g" configure.ac || die
+	fi
 	if [[ ${PV} == "9999" ]] ; then
 		eautoreconf
 	fi
@@ -44,6 +48,9 @@ src_configure() {
 	case ${CHOST} in
 	*-gnu*|*-uclibc*) myconf+=( "--with-wordbounds" ) ;; #467848
 	esac
+	if [[ "${CHOST}" =~ "muslx32" ]] ; then
+		myconf+=( "--disable-color" ) #temporary
+	fi
 	econf \
 		--bindir="${EPREFIX}"/bin \
 		--htmldir=/trash \
