diff -urp libjpeg-turbo-1.5.0.orig/simd/jchuff-sse2-64.asm libjpeg-turbo-1.5.0/simd/jchuff-sse2-64.asm
--- libjpeg-turbo-1.5.0.orig/simd/jchuff-sse2-64.asm	2016-10-21 21:02:15.729712214 -0700
+++ libjpeg-turbo-1.5.0/simd/jchuff-sse2-64.asm	2016-10-21 21:04:49.487559671 -0700
@@ -212,8 +212,8 @@ EXTN(jsimd_huff_encode_one_block_sse2):
 %endif
         mov buffer, r11  ; r11 is now sratch
 
-        mov put_buffer, MMWORD [r10+16]  ; put_buffer = state->cur.put_buffer;
-        mov put_bits,    DWORD [r10+24]  ; put_bits = state->cur.put_bits;
+        mov put_buffer, MMWORD [r10+SIZEOF_SIZE_T+SIZEOF_POINTER]  ; put_buffer = state->cur.put_buffer;
+        mov put_bits,    DWORD [r10+SIZEOF_SIZE_T+SIZEOF_POINTER+SIZEOF_QWORD]  ; put_bits = state->cur.put_bits;
         push r10  ; r10 is now scratch
 
         ; Encode the DC coefficient difference per section F.1.2.1
@@ -345,8 +345,8 @@ EXTN(jsimd_huff_encode_one_block_sse2):
 .EFN:
         pop r10
         ; Save put_buffer & put_bits
-        mov MMWORD [r10+16], put_buffer  ; state->cur.put_buffer = put_buffer;
-        mov DWORD  [r10+24], put_bits  ; state->cur.put_bits = put_bits;
+        mov MMWORD [r10+SIZEOF_SIZE_T+SIZEOF_POINTER], put_buffer  ; state->cur.put_buffer = put_buffer;
+        mov DWORD  [r10+SIZEOF_SIZE_T+SIZEOF_POINTER+SIZEOF_QWORD], put_bits  ; state->cur.put_bits = put_bits;
 
         pop rbx
 %ifdef WIN64
