diff -ur webkitgtk-2.12.3.orig/Source/WTF/wtf/Platform.h webkitgtk-2.12.3/Source/WTF/wtf/Platform.h
--- webkitgtk-2.12.3.orig/Source/WTF/wtf/Platform.h	2016-07-04 11:49:51.797458033 -0700
+++ webkitgtk-2.12.3/Source/WTF/wtf/Platform.h	2016-07-04 12:03:26.421466508 -0700
@@ -696,7 +696,7 @@
 #endif
 
 #if !defined(USE_JSVALUE64) && !defined(USE_JSVALUE32_64)
-#if (CPU(X86_64) && (OS(UNIX) || OS(WINDOWS))) \
+#if (CPU(X86_64) && (OS(UNIX) || OS(WINDOWS))) \
     || (CPU(IA64) && !CPU(IA64_32)) \
     || CPU(ALPHA) \
     || CPU(ARM64) \
@@ -710,6 +710,11 @@
 #endif
 #endif /* !defined(USE_JSVALUE64) && !defined(USE_JSVALUE32_64) */
 
+/* Disable JIT on x32 */
+#if CPU(X32)
+#define ENABLE_JIT 0
+#endif
+
 /* The JIT is enabled by default on all x86, x86-64, ARM & MIPS platforms except ARMv7k. */
 #if !defined(ENABLE_JIT) \
     && (CPU(X86) || CPU(X86_64) || CPU(ARM) || CPU(ARM64) || CPU(MIPS)) \
