diff -ur webkitgtk-2.12.3.orig/Source/JavaScriptCore/runtime/JSObject.h webkitgtk-2.12.3/Source/JavaScriptCore/runtime/JSObject.h
--- webkitgtk-2.12.3.orig/Source/JavaScriptCore/runtime/JSObject.h	2016-07-05 13:36:20.699423430 -0700
+++ webkitgtk-2.12.3/Source/JavaScriptCore/runtime/JSObject.h	2016-07-05 13:45:04.833428883 -0700
@@ -1493,7 +1493,7 @@
     return static_cast<size_t>(addressOffset);
 }
 
-COMPILE_ASSERT(!(sizeof(JSObject) % sizeof(WriteBarrierBase<Unknown>)), JSObject_inline_storage_has_correct_alignment);
+//COMPILE_ASSERT(!(sizeof(JSObject) % sizeof(WriteBarrierBase<Unknown>)), JSObject_inline_storage_has_correct_alignment);
 
 ALWAYS_INLINE Identifier makeIdentifier(VM& vm, const char* name)
 {
