Updated from muslx32 overlay
----
--- ./third_party/WebKit/Source/platform/wtf/StackUtil.cpp.orig	2018-05-27 18:30:40.729759606 -0700
+++ ./third_party/WebKit/Source/platform/wtf/StackUtil.cpp	2018-05-27 18:41:58.702766659 -0700
@@ -28,7 +28,7 @@ size_t GetUnderestimatedStackSize() {
 // FIXME: On Mac OSX and Linux, this method cannot estimate stack size
 // correctly for the main thread.
 
-#if defined(__GLIBC__) || defined(OS_ANDROID) || defined(OS_FREEBSD) || \
+#if defined(OS_LINUX) || defined(OS_ANDROID) || defined(OS_FREEBSD) || \
     defined(OS_FUCHSIA)
   // pthread_getattr_np() can fail if the thread is not invoked by
   // pthread_create() (e.g., the main thread of webkit_unit_tests).
@@ -96,7 +96,7 @@ size_t GetUnderestimatedStackSize() {
 }
 
 void* GetStackStart() {
-#if defined(__GLIBC__) || defined(OS_ANDROID) || defined(OS_FREEBSD) || \
+#if defined(OS_LINUX) || defined(OS_ANDROID) || defined(OS_FREEBSD) || \
     defined(OS_FUCHSIA)
   pthread_attr_t attr;
   int error;
--- base/threading/platform_thread_linux.cc.orig	2018-05-10 15:06:23.000000000 -0700
+++ base/threading/platform_thread_linux.cc	2018-05-27 16:28:10.336683130 -0700
@@ -173,7 +173,7 @@ void TerminateOnThread() {}
 
 size_t GetDefaultThreadStackSize(const pthread_attr_t& attributes) {
 #if !defined(THREAD_SANITIZER)
-  return 0;
+  return 1 << 21;
 #else
   // ThreadSanitizer bloats the stack heavily. Evidence has been that the
   // default stack size isn't enough for some browser tests.
