diff -urp firefox-49.0.orig/media/libvpx/third_party/x86inc/x86inc.asm firefox-49.0/media/libvpx/third_party/x86inc/x86inc.asm
--- firefox-49.0.orig/media/libvpx/third_party/x86inc/x86inc.asm	2016-05-12 10:13:17.000000000 -0700
+++ firefox-49.0/media/libvpx/third_party/x86inc/x86inc.asm	2016-10-23 10:48:58.400296283 -0700
@@ -59,6 +59,8 @@
     %define mangle(x) x
 %elifidn __OUTPUT_FORMAT__,elf64
     %define mangle(x) x
+%elifidn __OUTPUT_FORMAT__,elfx32
+    %define mangle(x) x
 %elifidn __OUTPUT_FORMAT__,elf
     %define mangle(x) x
 %elifidn __OUTPUT_FORMAT__,x64
@@ -621,6 +623,8 @@ DECLARE_ARG 7, 8, 9, 10, 11, 12, 13, 14
             global %1:function hidden
         %elifidn __OUTPUT_FORMAT__,elf64
             global %1:function hidden
+        %elifidn __OUTPUT_FORMAT__,elfx32
+            global %1:function hidden
         %elifidn __OUTPUT_FORMAT__,macho32
             %ifdef __NASM_VER__
                 global %1
@@ -675,6 +679,8 @@ SECTION .note.GNU-stack noalloc noexec n
 SECTION .note.GNU-stack noalloc noexec nowrite progbits
 %elifidn __OUTPUT_FORMAT__,elf64
 SECTION .note.GNU-stack noalloc noexec nowrite progbits
+%elifidn __OUTPUT_FORMAT__,elfx32
+SECTION .note.GNU-stack noalloc noexec nowrite progbits
 %endif
 
 ; cpuflags
Only in firefox-49.0/media/libvpx/third_party/x86inc: x86inc.asm.orig
Only in firefox-49.0/media/libvpx/third_party/x86inc: x86inc.asm.rej
diff -urp firefox-49.0.orig/media/libyuv/source/x86inc.asm firefox-49.0/media/libyuv/source/x86inc.asm
--- firefox-49.0.orig/media/libyuv/source/x86inc.asm	2016-05-12 10:13:17.000000000 -0700
+++ firefox-49.0/media/libyuv/source/x86inc.asm	2016-10-23 10:45:28.517399900 -0700
@@ -556,6 +556,9 @@ section .note.GNU-stack noalloc noexec n
 %ifidn __OUTPUT_FORMAT__,elf64
 section .note.GNU-stack noalloc noexec nowrite progbits
 %endif
+%ifidn __OUTPUT_FORMAT__,elfx32
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
 
 ; cpuflags
 
Only in firefox-49.0/media: p1.patch
Only in firefox-49.0/media: p2.patch
