--- firefox-52.1.0esr.orig/toolkit/crashreporter/google-breakpad/src/common/linux/memory_mapped_file.cc	2017-05-28 19:44:27.203476313 +0000
+++ firefox-52.1.0esr/toolkit/crashreporter/google-breakpad/src/common/linux/memory_mapped_file.cc	2017-05-28 19:56:26.810027817 +0000
@@ -83,7 +83,7 @@ bool MemoryMappedFile::Map(const char* p
   if (sys_fstat(fd, (struct stat*)&st) == -1 || st.st_size < 0) {
 #else
   struct kernel_stat64 st;
-  if (sys_fstat64(fd, &st) == -1 || st.st_size < 0) {
+  if (sys_fstat64(fd, (struct stat64*)&st) == -1 || st.st_size < 0) {
 #endif
     sys_close(fd);
     return false;
