diff -ur firefox-45.2.0esr.orig/js/src/jit/shared/Assembler-shared.h firefox-45.2.0esr/js/src/jit/shared/Assembler-shared.h
--- firefox-45.2.0esr.orig/js/src/jit/shared/Assembler-shared.h	2016-05-12 10:09:59.000000000 -0700
+++ firefox-45.2.0esr/js/src/jit/shared/Assembler-shared.h	2016-09-04 18:00:30.322347592 -0700
@@ -676,8 +676,16 @@
     // The return address pushed by the call (in the case of ARM/MIPS the return
     // address is pushed by the first instruction of the prologue).
     void* returnAddress;
+
+#if defined(__x86_64__) && defined(__ILP32__)
+    void* padding[2];
+#endif
 };
+#if defined(__x86_64__) && defined(__ILP32__)
+static_assert(sizeof(AsmJSFrame) == 4 * sizeof(void*), "?!");
+#else
 static_assert(sizeof(AsmJSFrame) == 2 * sizeof(void*), "?!");
+#endif
 static const uint32_t AsmJSFrameBytesAfterReturnAddress = sizeof(void*);
 
 struct AsmJSGlobalAccess
