diff -urp firefox-52.1.0esr.orig/security/nss/lib/freebl/rijndael.c firefox-52.1.0esr/security/nss/lib/freebl/rijndael.c
--- firefox-52.1.0esr.orig/security/nss/lib/freebl/rijndael.c	2017-04-11 02:13:26.000000000 +0000
+++ firefox-52.1.0esr/security/nss/lib/freebl/rijndael.c	2017-05-28 03:55:40.267318204 +0000
@@ -1095,6 +1095,9 @@ aes_InitContext(AESContext *cx, const un
         }
     }
     use_hw_aes = (PRBool)(has_intel_aes > 0 && (keysize % 8) == 0 && blocksize == 16);
+#if defined(__x86_64__) && defined(__ILP32__)
+    use_hw_aes = 0; // disable on x32 for now
+#endif
 #ifdef INTEL_GCM
     use_hw_gcm = (PRBool)(use_hw_aes && has_intel_avx > 0 && has_intel_clmul > 0);
 #endif
