diff -urp libjpeg-turbo-1.5.0.orig/simd/jsimdext.inc libjpeg-turbo-1.5.0/simd/jsimdext.inc
--- libjpeg-turbo-1.5.0.orig/simd/jsimdext.inc	2016-06-07 10:33:40.000000000 -0700
+++ libjpeg-turbo-1.5.0/simd/jsimdext.inc	2016-10-20 13:43:34.261116661 -0700
@@ -131,7 +131,13 @@ section .note.GNU-stack noalloc noexec n
 ;  Common types
 ;
 %ifdef __x86_64__
+%ifdef __ILP32__
+%define POINTER                 dword           ; general pointer type
+%define SIZEOF_POINTER          SIZEOF_DWORD    ; sizeof(POINTER)
+%define POINTER_BIT             DWORD_BIT       ; sizeof(POINTER)*BYTE_BIT
+%else
 %define POINTER                 qword           ; general pointer type
 %define SIZEOF_POINTER          SIZEOF_QWORD    ; sizeof(POINTER)
 %define POINTER_BIT             QWORD_BIT       ; sizeof(POINTER)*BYTE_BIT
+%endif
 %else
diff -urp libjpeg-turbo-1.5.0.orig/simd/jsimdext.inc libjpeg-turbo-1.5.0/simd/jsimdext.inc
--- libjpeg-turbo-1.5.0.orig/simd/jsimdext.inc	2016-10-21 14:08:22.212763617 -0700
+++ libjpeg-turbo-1.5.0/simd/jsimdext.inc	2016-10-21 15:25:57.272243358 -0700
@@ -145,6 +145,7 @@ section .note.GNU-stack noalloc noexec n
 %define SIZEOF_POINTER          SIZEOF_DWORD    ; sizeof(POINTER)
 %define POINTER_BIT             DWORD_BIT       ; sizeof(POINTER)*BYTE_BIT
 %endif
+%define SIZEOF_SIZE_T           SIZEOF_POINTER
 
 %define INT                     dword           ; signed integer type
 %define SIZEOF_INT              SIZEOF_DWORD    ; sizeof(INT)
