diff -ur firefox-47.0.1.orig/toolkit/crashreporter/google-breakpad/src/third_party/lss/linux_syscall_support.h firefox-47.0.1/toolkit/crashreporter/google-breakpad/src/third_party/lss/linux_syscall_support.h
--- firefox-47.0.1.orig/toolkit/crashreporter/google-breakpad/src/third_party/lss/linux_syscall_support.h	2016-07-30 21:14:32.597159904 -0700
+++ firefox-47.0.1/toolkit/crashreporter/google-breakpad/src/third_party/lss/linux_syscall_support.h	2016-07-30 21:20:41.173172050 -0700
@@ -85,7 +85,7 @@
 /* We currently only support x86-32, x86-64, ARM, MIPS, and PPC on Linux.
  * Porting to other related platforms should not be difficult.
  */
-#if (defined(__i386__) || defined(__x86_64__) || defined(__ARM_ARCH_3__) ||   \
+#if (defined(__i386__) || defined(__x86_64__) && defined(__LP64__) || defined(__ARM_ARCH_3__) ||   \
      defined(__mips__) || defined(__PPC__) || defined(__ARM_EABI__) || \
      defined(__aarch64__)) \
   && (defined(__linux) || defined(__ANDROID__))
