diff -ur firefox-47.0.1.orig/media/ffvpx/libavutil/x86/moz.build firefox-47.0.1/media/ffvpx/libavutil/x86/moz.build
--- firefox-47.0.1.orig/media/ffvpx/libavutil/x86/moz.build	2016-07-31 10:09:47.268692675 -0700
+++ firefox-47.0.1/media/ffvpx/libavutil/x86/moz.build	2016-07-31 10:06:57.309687075 -0700
@@ -4,6 +4,9 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+include('/media/ffvpx/ffvpxcommon.mozbuild')
+CFLAGS += ['-DHAVE_YASM=0', '-DHAVE_INLINE_ASM=0']
+
 SOURCES += [
     'cpu.c',
     'cpuid.asm',
@@ -18,4 +20,3 @@
 
 FINAL_LIBRARY = 'mozavutil'
 
-include('/media/ffvpx/ffvpxcommon.mozbuild')
