--- firefox-47.0.1.orig/toolkit/crashreporter/google-breakpad/src/common/linux/memory_mapped_file.cc	2016-07-30 22:39:06.896327115 -0700
+++ firefox-47.0.1/toolkit/crashreporter/google-breakpad/src/common/linux/memory_mapped_file.cc	2016-07-30 22:46:15.789341248 -0700
@@ -38,6 +38,7 @@
 #include <sys/stat.h>
 #endif
 #include <unistd.h>
+#include <sys/stat.h>
 
 #include "common/memory_range.h"
 #include "third_party/lss/linux_syscall_support.h"
