diff -ur firefox-45.2.0esr.orig/js/src/configure.in firefox-45.2.0esr/js/src/configure.in
--- firefox-45.2.0esr.orig/js/src/configure.in	2016-08-01 03:40:21.936769821 -0700
+++ firefox-45.2.0esr/js/src/configure.in	2016-08-01 10:52:05.182623396 -0700
@@ -1402,6 +1402,12 @@
 if test "$result" = "yes"; then
     AC_DEFINE(HAVE_64BIT_BUILD)
     HAVE_64BIT_BUILD=1
+else
+    AC_EGREP_CPP(x32_test,
+    [#if defined(__x86_64__) && defined (__ILP32__)
+        x32_test
+        #endif
+    ], HAVE_64BIT_BUILD=1, HAVE_64BIT_BUILD=)
 fi
 AC_SUBST(HAVE_64BIT_BUILD)
 AC_LANG_RESTORE
diff -ur firefox-45.2.0esr.orig/js/src/jit/x64/Assembler-x64.cpp firefox-45.2.0esr/js/src/jit/x64/Assembler-x64.cpp
--- firefox-45.2.0esr.orig/js/src/jit/x64/Assembler-x64.cpp	2016-05-12 10:09:59.000000000 -0700
+++ firefox-45.2.0esr/js/src/jit/x64/Assembler-x64.cpp	2016-08-01 05:14:24.178955746 -0700
@@ -39,7 +39,11 @@
             stackOffset_ += Simd128DataSize;
         } else {
             current_ = ABIArg(stackOffset_);
+#if defined(__ILP32__)
+            stackOffset_ += sizeof(uint32_t);
+#else
             stackOffset_ += sizeof(uint64_t);
+#endif
         }
         return current_;
     }
@@ -71,7 +75,11 @@
       case MIRType_Pointer:
         if (intRegIndex_ == NumIntArgRegs) {
             current_ = ABIArg(stackOffset_);
+#if defined(__ILP32__)
+            stackOffset_ += sizeof(uint32_t);
+#else
             stackOffset_ += sizeof(uint64_t);
+#endif
             break;
         }
         current_ = ABIArg(IntArgRegs[intRegIndex_++]);
@@ -80,7 +88,11 @@
       case MIRType_Float32:
         if (floatRegIndex_ == NumFloatArgRegs) {
             current_ = ABIArg(stackOffset_);
+#if defined(__ILP32__)
+            stackOffset_ += sizeof(uint32_t);
+#else
             stackOffset_ += sizeof(uint64_t);
+#endif
             break;
         }
         if (type == MIRType_Float32)
