diff -ur firefox-45.2.0esr.orig/media/libav/libavcommon.mozbuild firefox-45.2.0esr/media/libav/libavcommon.mozbuild
--- firefox-45.2.0esr.orig/media/libav/libavcommon.mozbuild	2016-05-12 10:09:56.000000000 -0700
+++ firefox-45.2.0esr/media/libav/libavcommon.mozbuild	2016-07-30 17:52:52.769761185 -0700
@@ -24,7 +24,7 @@
     ASFLAGS += ['-Pconfig_darwin.asm', '-DPREFIX']
 else:
     # Default to unix, similar to how ASFLAGS setup works in configure.in
-    ASFLAGS += ['-Pconfig_unix.asm']
+    ASFLAGS += ['-Pconfig_unix.asm', '-f elfx32']
 
 LOCAL_INCLUDES += ['/media/libav']
 
