diff -ur firefox-45.2.0esr.orig/js/src/jit/x64/Assembler-x64.h firefox-45.2.0esr/js/src/jit/x64/Assembler-x64.h
--- firefox-45.2.0esr.orig/js/src/jit/x64/Assembler-x64.h	2016-05-12 10:09:59.000000000 -0700
+++ firefox-45.2.0esr/js/src/jit/x64/Assembler-x64.h	2016-08-01 13:49:32.219974241 -0700
@@ -191,7 +191,11 @@
 
 static MOZ_CONSTEXPR_VAR Register PreBarrierReg = rdx;
 
+#if defined(__ILP32__)
+static MOZ_CONSTEXPR_VAR uint32_t ABIStackAlignment = 8;
+#else
 static MOZ_CONSTEXPR_VAR uint32_t ABIStackAlignment = 16;
+#endif
 static MOZ_CONSTEXPR_VAR uint32_t CodeAlignment = 16;
 static MOZ_CONSTEXPR_VAR uint32_t JitStackAlignment = 16;
 
@@ -256,8 +260,13 @@
     //    +2 bytes for ud2 instruction
     //    +8 bytes for 64-bit address
     //
+#if defined(__ILP32__)
+    static const uint32_t SizeOfExtendedJump = 1 + 1 + 4 + 2 + 4;
+    static const uint32_t SizeOfJumpTableEntry = 12;
+#else
     static const uint32_t SizeOfExtendedJump = 1 + 1 + 4 + 2 + 8;
     static const uint32_t SizeOfJumpTableEntry = 16;
+#endif
 
     uint32_t extendedJumpTable_;
 
