--- mozilla-esr38.orig/tools/profiler/platform-linux.cc	2016-06-25 15:17:22.880781513 -0700
+++ mozilla-esr38/tools/profiler/platform-linux.cc	2016-06-25 15:18:24.226779549 -0700
@@ -651,11 +651,13 @@
 void TickSample::PopulateContext(void* aContext)
 {
   MOZ_ASSERT(aContext);
+  #if defined(__GLIBC__)
   ucontext_t* pContext = reinterpret_cast<ucontext_t*>(aContext);
   if (!getcontext(pContext)) {
     context = pContext;
     SetSampleContext(this, aContext);
   }
+  #endif
 }
 
 void OS::SleepMicro(int microseconds)
