diff -ur firefox-45.2.0esr.orig/toolkit/crashreporter/google-breakpad/src/common/linux/memory_mapped_file.cc firefox-45.2.0esr/toolkit/crashreporter/google-breakpad/src/common/linux/memory_mapped_file.cc
--- firefox-45.2.0esr.orig/toolkit/crashreporter/google-breakpad/src/common/linux/memory_mapped_file.cc	2016-05-12 10:10:05.000000000 -0700
+++ firefox-45.2.0esr/toolkit/crashreporter/google-breakpad/src/common/linux/memory_mapped_file.cc	2016-07-30 06:00:24.524352527 -0700
@@ -38,6 +38,7 @@
 #include <sys/stat.h>
 #endif
 #include <unistd.h>
+#include <asm-generic/unistd.h>
 
 #include "common/memory_range.h"
 #include "third_party/lss/linux_syscall_support.h"
