--- firefox-52.1.0esr.orig/js/src/jsmath.cpp	2017-04-11 02:13:16.000000000 +0000
+++ firefox-52.1.0esr/js/src/jsmath.cpp	2017-05-28 10:48:58.081643406 +0000
@@ -63,7 +63,11 @@
 // Older glibc versions don't define SYS_getrandom, so we define it here if
 // it's not available. See bug 995069.
 # if defined(__x86_64__)
-#  define GETRANDOM_NR 318
+#  if defined(__ILP32__)
+#    define GETRANDOM_NR 0x40000000 + 318
+#  else
+#    define GETRANDOM_NR 318
+#  endif
 # elif defined(__i386__)
 #  define GETRANDOM_NR 355
 # elif defined(__arm__)
