diff -urp firefox-52.1.0esr.orig/old-configure firefox-52.1.0esr/old-configure
--- firefox-52.1.0esr.orig/old-configure	2017-04-17 18:33:03.000000000 +0000
+++ firefox-52.1.0esr/old-configure	2017-05-28 03:28:36.153264686 +0000
@@ -15189,6 +15189,10 @@ i?86-apple-darwin*|x86_64-apple-darwin*)
     MOZ_CRASHREPORTER=1
   fi
   ;;
+x86_64-*-linux-gnux32*| x86_64-*-linux-muslx32*)
+  # x32 is not supported.
+  MOZ_CRASHREPORTER=
+  ;;
 i?86-*-linux*|x86_64-*-linux*|arm-*-linux*)
   if test "$MOZ_ENABLE_GTK"; then
     MOZ_CRASHREPORTER=1
diff -urp firefox-52.1.0esr.orig/old-configure.in firefox-52.1.0esr/old-configure.in
--- firefox-52.1.0esr.orig/old-configure.in	2017-05-28 03:26:44.295261000 +0000
+++ firefox-52.1.0esr/old-configure.in	2017-05-28 03:28:36.155264686 +0000
@@ -3530,6 +3530,10 @@ i?86-apple-darwin*|x86_64-apple-darwin*)
     MOZ_CRASHREPORTER=1
   fi
   ;;
+x86_64-*-linux-gnux32*| x86_64-*-linux-muslx32*)
+  # x32 is not supported.
+  MOZ_CRASHREPORTER=
+  ;;
 i?86-*-linux*|x86_64-*-linux*|arm-*-linux*)
   if test "$MOZ_ENABLE_GTK"; then
     MOZ_CRASHREPORTER=1
diff -urp firefox-52.1.0esr.orig/toolkit/crashreporter/google-breakpad/src/third_party/lss/linux_syscall_support.h firefox-52.1.0esr/toolkit/crashreporter/google-breakpad/src/third_party/lss/linux_syscall_support.h
--- firefox-52.1.0esr.orig/toolkit/crashreporter/google-breakpad/src/third_party/lss/linux_syscall_support.h	2017-05-28 03:26:44.320261001 +0000
+++ firefox-52.1.0esr/toolkit/crashreporter/google-breakpad/src/third_party/lss/linux_syscall_support.h	2017-05-28 03:28:36.158264686 +0000
@@ -86,7 +86,7 @@
  * on Linux.
  * Porting to other related platforms should not be difficult.
  */
-#if (defined(__i386__) || defined(__x86_64__) || defined(__ARM_ARCH_3__) ||   \
+#if (defined(__i386__) || defined(__x86_64__) && defined(__LP64__) || defined(__ARM_ARCH_3__) ||   \
      defined(__mips__) || defined(__PPC__) || defined(__ARM_EABI__) || \
      defined(__aarch64__) || defined(__s390__)) \
   && (defined(__linux) || defined(__ANDROID__))
