diff -urp firefox-52.1.0esr.orig/build/moz.configure/toolchain.configure firefox-52.1.0esr/build/moz.configure/toolchain.configure
--- firefox-52.1.0esr.orig/build/moz.configure/toolchain.configure	2017-05-28 16:45:10.993383260 +0000
+++ firefox-52.1.0esr/build/moz.configure/toolchain.configure	2017-05-28 16:44:18.098341690 +0000
@@ -801,10 +801,10 @@ include('compile-checks.configure')
 @depends(have_64_bit,
          try_compile(body='static_assert(sizeof(void *) == 8, "")',
                      check_msg='for 64-bit OS'))
-def check_have_64_bit(have_64_bit, compiler_have_64_bit):
-    if have_64_bit != compiler_have_64_bit:
-        configure_error('The target compiler does not agree with configure '
-                        'about the target bitness.')
+def check_have_64_bit(have_64_bit, compiler_have_64_bit):
+    if have_64_bit != compiler_have_64_bit:
+        return
+
 
 
 @depends(c_compiler)
