diff -ur firefox-49.0.orig/js/src/threading/posix/Thread.cpp firefox-49.0/js/src/threading/posix/Thread.cpp
--- firefox-49.0.orig/js/src/threading/posix/Thread.cpp	2016-09-05 13:12:56.000000000 -0700
+++ firefox-49.0/js/src/threading/posix/Thread.cpp	2016-09-28 18:23:51.545404584 -0700
@@ -145,6 +145,7 @@
   name = nameBuf;
 #endif
 
+#if defined(__GLIBC__)
   int rv;
 #ifdef XP_DARWIN
   rv = pthread_setname_np(name);
@@ -157,4 +158,5 @@
   rv = pthread_setname_np(pthread_self(), name);
 #endif
   MOZ_RELEASE_ASSERT(!rv);
+#endif
 }
