diff -urp firefox-52.1.0esr.orig/toolkit/crashreporter/google-breakpad/src/common/memory.h firefox-52.1.0esr/toolkit/crashreporter/google-breakpad/src/common/memory.h
--- firefox-52.1.0esr.orig/toolkit/crashreporter/google-breakpad/src/common/memory.h	2017-04-11 02:13:23.000000000 +0000
+++ firefox-52.1.0esr/toolkit/crashreporter/google-breakpad/src/common/memory.h	2017-05-28 19:32:06.032657240 +0000
@@ -50,6 +50,12 @@
 #include "third_party/lss/linux_syscall_support.h"
 #endif
 
+#ifndef __GLIBC__
+#define sys_mmap mmap
+#define sys_munmap munmap
+#define sys_fstat64 fstat
+#endif
+
 namespace google_breakpad {
 
 // This is very simple allocator which fetches pages from the kernel directly.
