diff -urp firefox-49.0.orig/media/ffvpx/libavutil/x86/x86inc.asm firefox-49.0/media/ffvpx/libavutil/x86/x86inc.asm
--- firefox-49.0.orig/media/ffvpx/libavutil/x86/x86inc.asm	2016-05-31 21:11:43.000000000 -0700
+++ firefox-49.0/media/ffvpx/libavutil/x86/x86inc.asm	2016-10-02 20:17:22.031594007 -0700
@@ -226,6 +226,17 @@ DECLARE_REG_TMP_SIZE 0,1,2,3,4,5,6,7,8,9
     %define gprsize 4
 %endif
 
+%if ARCH_X86_64
+%ifdef __ILP32__
+    %define ARCH_X86_64_X32 1
+    %define ptrsize 4
+%else
+    %define ptrsize 8
+%endif
+%else
+    %define ptrsize 4
+%endif
+
 %macro PUSH 1
     push %1
     %ifidn rstk, rsp
@@ -304,6 +315,7 @@ DECLARE_REG_TMP_SIZE 0,1,2,3,4,5,6,7,8,9
         %assign %%i 0
         %rep n_arg_names
             CAT_UNDEF arg_name %+ %%i, q
+            CAT_UNDEF arg_name %+ %%i, p
             CAT_UNDEF arg_name %+ %%i, d
             CAT_UNDEF arg_name %+ %%i, w
             CAT_UNDEF arg_name %+ %%i, h
@@ -320,6 +332,7 @@ DECLARE_REG_TMP_SIZE 0,1,2,3,4,5,6,7,8,9
     %assign %%i 0
     %rep %0
         %xdefine %1q r %+ %%i %+ q
+        %xdefine %1p r %+ %%i %+ p
         %xdefine %1d r %+ %%i %+ d
         %xdefine %1w r %+ %%i %+ w
         %xdefine %1h r %+ %%i %+ h
@@ -1483,6 +1496,32 @@ FMA_INSTR pmadcswd, pmaddwd, paddd
 ; This lets us use tzcnt without bumping the yasm version requirement yet.
 %define tzcnt rep bsf
 
+; FIXME : integrate into DECLARE_REG macros
+
+%ifdef ARCH_X86_64_X32
+%xdefine r0p r0d
+%xdefine r1p r1d
+%xdefine r2p r2d
+%xdefine r3p r3d
+%xdefine r4p r4d
+%xdefine r5p r5d
+%xdefine r6p r6d
+%xdefine r7p r7d
+%xdefine r8p r8d
+%xdefine r9p r9d
+%else
+%xdefine r0p r0
+%xdefine r1p r1
+%xdefine r2p r2
+%xdefine r3p r3
+%xdefine r4p r4
+%xdefine r5p r5
+%xdefine r6p r6
+%xdefine r7p r7
+%xdefine r8p r8
+%xdefine r9p r9
+%endif
+
 ; Macros for consolidating FMA3 and FMA4 using 4-operand (dst, src1, src2, src3) syntax.
 ; FMA3 is only possible if dst is the same as one of the src registers.
 ; Either src2 or src3 can be a memory operand.
