diff -urp webkitgtk-9999.133571.1.orig/Source/JavaScriptCore/interpreter/Interpreter.cpp webkitgtk-9999.133571.1/Source/JavaScriptCore/interpreter/Interpreter.cpp
--- webkitgtk-9999.133571.1.orig/Source/JavaScriptCore/interpreter/Interpreter.cpp	2017-06-10 10:20:30.477177995 -0700
+++ webkitgtk-9999.133571.1/Source/JavaScriptCore/interpreter/Interpreter.cpp	2017-06-10 10:32:08.774201005 -0700
@@ -119,8 +119,8 @@ Interpreter::StackPolicy::StackPolicy(In
     const size_t size = stack.size();
 
     const size_t DEFAULT_REQUIRED_STACK = 1024 * 1024;
-    const size_t DEFAULT_MINIMUM_USEABLE_STACK = 128 * 1024;
-    const size_t DEFAULT_ERROR_MODE_REQUIRED_STACK = 32 * 1024;
+    const size_t DEFAULT_MINIMUM_USEABLE_STACK = 32 * 1024;
+    const size_t DEFAULT_ERROR_MODE_REQUIRED_STACK = 16 * 1024;
 
     // Here's the policy in a nutshell:
     //
Only in webkitgtk-9999.133571.1/Source/JavaScriptCore/interpreter: Interpreter.cpp.orig
Only in webkitgtk-9999.133571.1/Source/JavaScriptCore/interpreter: Interpreter.cpp.rej
