diff -ur webkitgtk-2.12.3.orig/Source/JavaScriptCore/b3/B3ConstPtrValue.h webkitgtk-2.12.3/Source/JavaScriptCore/b3/B3ConstPtrValue.h
--- webkitgtk-2.12.3.orig/Source/JavaScriptCore/b3/B3ConstPtrValue.h	2016-04-10 23:18:43.000000000 -0700
+++ webkitgtk-2.12.3/Source/JavaScriptCore/b3/B3ConstPtrValue.h	2016-07-25 22:28:37.339008629 -0700
@@ -37,7 +37,7 @@
 // platform-agnostic code. Note that a ConstPtrValue will behave like either a Const32Value or
 // Const64Value depending on platform.
 
-#if USE(JSVALUE64)
+#if USE(JSVALUE64) && !CPU(X32)
 typedef Const64Value ConstPtrValueBase;
 #else
 typedef Const32Value ConstPtrValueBase;
