diff -ur webkitgtk-2.12.3.orig/Source/JavaScriptCore/offlineasm/x86.rb webkitgtk-2.12.3/Source/JavaScriptCore/offlineasm/x86.rb
--- webkitgtk-2.12.3.orig/Source/JavaScriptCore/offlineasm/x86.rb	2016-07-25 00:22:29.953143332 -0700
+++ webkitgtk-2.12.3/Source/JavaScriptCore/offlineasm/x86.rb	2016-07-25 00:25:34.106131449 -0700
@@ -124,6 +124,8 @@
         false
     when "X86_64_WIN"
         true
+    when "X32"
+	false
     else
         raise "bad value for $activeBackend: #{$activeBackend}"
     end
@@ -139,6 +141,8 @@
         false
     when "X86_64_WIN"
         false
+    when "X32"
+	false
     else
         raise "bad value for $activeBackend: #{$activeBackend}"
     end
