diff -ur webkitgtk-2.12.3.orig/Source/JavaScriptCore/llint/LLIntOfflineAsmConfig.h webkitgtk-2.12.3/Source/JavaScriptCore/llint/LLIntOfflineAsmConfig.h
--- webkitgtk-2.12.3.orig/Source/JavaScriptCore/llint/LLIntOfflineAsmConfig.h	2016-07-27 17:26:51.795168479 -0700
+++ webkitgtk-2.12.3/Source/JavaScriptCore/llint/LLIntOfflineAsmConfig.h	2016-07-27 17:39:05.006192640 -0700
@@ -105,7 +105,7 @@
 #define OFFLINE_ASM_X86_64 0
 #endif
 
-#if CPU(X86_64) && PLATFORM(WIN)
+#if CPU(X86_64) && !CPU(X32) && PLATFORM(WIN)
 #define OFFLINE_ASM_X86_64_WIN 1
 #else
 #define OFFLINE_ASM_X86_64_WIN 0
