diff -urp webkitgtk-9999.131899.1.orig/Source/JavaScriptCore/offlineasm/cloop.rb webkitgtk-9999.131899.1/Source/JavaScriptCore/offlineasm/cloop.rb
--- webkitgtk-9999.131899.1.orig/Source/JavaScriptCore/offlineasm/cloop.rb	2017-06-09 17:54:00.390227538 -0700
+++ webkitgtk-9999.131899.1/Source/JavaScriptCore/offlineasm/cloop.rb	2017-06-09 18:09:17.704257766 -0700
@@ -353,7 +353,7 @@ def cloopEmitShiftOperation(operands, ty
 end
 
 def cloopEmitCompareDoubleWithNaNCheckAndBranch(operands, condition)
-    $asm.putc "if (isnan(#{operands[0].clValue(:double)}) || isnan(#{operands[1].clValue(:double)})"
+    $asm.putc "if (isnan(#{operands[0].clValue(:double)})"
     $asm.putc "    || (#{operands[0].clValue(:double)} #{condition} #{operands[1].clValue(:double)}))"
     $asm.putc "    goto #{operands[2].cLabel};"
 end
Only in webkitgtk-9999.131899.1/Source/JavaScriptCore/offlineasm: cloop.rb.orig
Only in webkitgtk-9999.131899.1/Source/JavaScriptCore/offlineasm: cloop.rb.rej
