diff -urp webkitgtk-2.0.4.orig/Source/JavaScriptCore/interpreter/Interpreter.cpp webkitgtk-2.0.4/Source/JavaScriptCore/interpreter/Interpreter.cpp
--- webkitgtk-2.0.4.orig/Source/JavaScriptCore/interpreter/Interpreter.cpp	2013-07-19 00:03:48.000000000 -0700
+++ webkitgtk-2.0.4/Source/JavaScriptCore/interpreter/Interpreter.cpp	2017-06-02 07:29:27.296001376 -0700
@@ -120,8 +120,8 @@ Interpreter::StackPolicy::StackPolicy(In
     const size_t size = stack.size();
 
     const size_t DEFAULT_REQUIRED_STACK = 1024 * 1024;
-    const size_t DEFAULT_MINIMUM_USEABLE_STACK = 128 * 1024;
-    const size_t DEFAULT_ERROR_MODE_REQUIRED_STACK = 32 * 1024;
+    const size_t DEFAULT_MINIMUM_USEABLE_STACK = 32 * 1024;
+    const size_t DEFAULT_ERROR_MODE_REQUIRED_STACK = 16 * 1024;
 
     // Here's the policy in a nutshell:
     //
