diff -urp webkitgtk-9999.148063.orig/Source/ThirdParty/ANGLE/src/compiler/glslang.l webkitgtk-9999.148063/Source/ThirdParty/ANGLE/src/compiler/glslang.l
--- webkitgtk-9999.148063.orig/Source/ThirdParty/ANGLE/src/compiler/glslang.l	2017-06-11 14:16:10.789790458 -0700
+++ webkitgtk-9999.148063/Source/ThirdParty/ANGLE/src/compiler/glslang.l	2017-06-11 18:31:43.551545011 -0700
@@ -40,7 +40,7 @@ WHICH GENERATES THE GLSL ES LEXER (glsla
 #include "compiler/ParseHelper.h"
 #include "compiler/preprocessor/Token.h"
 #include "compiler/util.h"
-#include "glslang_tab.h"
+#include "compiler/glslang_tab.h"
 
 /* windows only pragma */
 #ifdef _MSC_VER
diff -urp webkitgtk-9999.148063.orig/Source/ThirdParty/ANGLE/src/compiler/glslang_lex.cpp webkitgtk-9999.148063/Source/ThirdParty/ANGLE/src/compiler/glslang_lex.cpp
--- webkitgtk-9999.148063.orig/Source/ThirdParty/ANGLE/src/compiler/glslang_lex.cpp	2017-06-11 14:16:10.790790458 -0700
+++ webkitgtk-9999.148063/Source/ThirdParty/ANGLE/src/compiler/glslang_lex.cpp	2017-06-11 18:31:44.976544988 -0700
@@ -810,7 +810,7 @@ WHICH GENERATES THE GLSL ES LEXER (glsla
 #include "compiler/ParseHelper.h"
 #include "compiler/preprocessor/Token.h"
 #include "compiler/util.h"
-#include "glslang_tab.h"
+#include "compiler/glslang_tab.h"
 
 /* windows only pragma */
 #ifdef _MSC_VER
