diff -urp webkitgtk-9999.133571.1.orig/Source/WTF/wtf/Assertions.cpp webkitgtk-9999.133571.1/Source/WTF/wtf/Assertions.cpp
--- webkitgtk-9999.133571.1.orig/Source/WTF/wtf/Assertions.cpp	2017-06-09 19:59:59.061476614 -0700
+++ webkitgtk-9999.133571.1/Source/WTF/wtf/Assertions.cpp	2017-06-09 22:21:53.509757186 -0700
@@ -61,7 +61,8 @@
 #if (OS(DARWIN) || OS(LINUX)) && !OS(ANDROID)
 #include <cxxabi.h>
 #include <dlfcn.h>
-#include <execinfo.h>
+// removed for musl - we need better intelligence here
+//#include <execinfo.h>
 #endif
 
 #if OS(ANDROID)
@@ -243,7 +244,8 @@ void WTFReportArgumentAssertionFailure(c
 void WTFGetBacktrace(void** stack, int* size)
 {
 #if (OS(DARWIN) || OS(LINUX)) && !OS(ANDROID)
-    *size = backtrace(stack, *size);
+// removed for musl - we need better intelligence here
+//    *size = backtrace(stack, *size);
 #elif OS(WINDOWS) && !OS(WINCE)
     // The CaptureStackBackTrace function is available in XP, but it is not defined
     // in the Windows Server 2003 R2 Platform SDK. So, we'll grab the function
@@ -289,9 +291,10 @@ void WTFReportBacktrace()
 void WTFPrintBacktrace(void** stack, int size)
 {
 #if USE(BACKTRACE_SYMBOLS)
-    char** symbols = backtrace_symbols(stack, size);
-    if (!symbols)
-        return;
+// removed for musl - we need better intelligence here
+//    char** symbols = backtrace_symbols(stack, size);
+//    if (!symbols)
+//        return;
 #endif
 
     for (int i = 0; i < size; ++i) {
Only in webkitgtk-9999.133571.1/Source/WTF/wtf: Assertions.cpp.orig
Only in webkitgtk-9999.133571.1/Source/WTF/wtf: Assertions.cpp.rej
