diff -ur webkitgtk-2.12.3.orig/Source/JavaScriptCore/assembler/MaxFrameExtentForSlowPathCall.h webkitgtk-2.12.3/Source/JavaScriptCore/assembler/MaxFrameExtentForSlowPathCall.h
--- webkitgtk-2.12.3.orig/Source/JavaScriptCore/assembler/MaxFrameExtentForSlowPathCall.h	2016-04-10 23:18:43.000000000 -0700
+++ webkitgtk-2.12.3/Source/JavaScriptCore/assembler/MaxFrameExtentForSlowPathCall.h	2016-07-25 10:33:53.601776035 -0700
@@ -44,7 +44,7 @@
 // 4 args in registers, but stack space needs to be allocated for all args.
 static const size_t maxFrameExtentForSlowPathCall = 64;
 
-#elif CPU(X86_64)
+#elif CPU(X86_64) || CPU(X32)
 // All args in registers.
 static const size_t maxFrameExtentForSlowPathCall = 0;
 
