diff -urp webkitgtk-9999.148063.orig/Source/WTF/wtf/Assertions.cpp webkitgtk-9999.148063/Source/WTF/wtf/Assertions.cpp
--- webkitgtk-9999.148063.orig/Source/WTF/wtf/Assertions.cpp	2017-06-09 14:57:21.555878280 -0700
+++ webkitgtk-9999.148063/Source/WTF/wtf/Assertions.cpp	2017-06-09 15:01:11.354885853 -0700
@@ -64,7 +64,8 @@
 #if OS(DARWIN) || (OS(LINUX) && !defined(__UCLIBC__))
 #include <cxxabi.h>
 #include <dlfcn.h>
-#include <execinfo.h>
+// removed for musl - we need better intelligence here
+//#include <execinfo.h>
 #endif
 
 #if PLATFORM(BLACKBERRY)
@@ -240,7 +241,8 @@ void WTFReportArgumentAssertionFailure(c
 void WTFGetBacktrace(void** stack, int* size)
 {
 #if OS(DARWIN) || (OS(LINUX) && !defined(__UCLIBC__))
-    *size = backtrace(stack, *size);
+// removed for musl - we need better intelligence here
+//    *size = backtrace(stack, *size);
 #elif OS(WINDOWS) && !OS(WINCE)
     // The CaptureStackBackTrace function is available in XP, but it is not defined
     // in the Windows Server 2003 R2 Platform SDK. So, we'll grab the function
@@ -286,9 +288,10 @@ void WTFReportBacktrace()
 void WTFPrintBacktrace(void** stack, int size)
 {
 #if USE(BACKTRACE_SYMBOLS)
-    char** symbols = backtrace_symbols(stack, size);
-    if (!symbols)
-        return;
+// removed for musl - we need better intelligence here
+//    char** symbols = backtrace_symbols(stack, size);
+//    if (!symbols)
+//        return;
 #endif
 
     for (int i = 0; i < size; ++i) {
Only in webkitgtk-9999.148063/Source/WTF/wtf: Assertions.cpp.orig
Only in webkitgtk-9999.148063/Source/WTF/wtf: Assertions.cpp.rej
