diff -ur webkitgtk-2.12.3.orig/Source/JavaScriptCore/dfg/DFGSpeculativeJIT64.cpp webkitgtk-2.12.3/Source/JavaScriptCore/dfg/DFGSpeculativeJIT64.cpp
--- webkitgtk-2.12.3.orig/Source/JavaScriptCore/dfg/DFGSpeculativeJIT64.cpp	2016-07-26 00:47:22.066471412 -0700
+++ webkitgtk-2.12.3/Source/JavaScriptCore/dfg/DFGSpeculativeJIT64.cpp	2016-07-26 01:15:05.543364063 -0700
@@ -829,7 +829,8 @@
     CallLinkInfo* callLinkInfo = m_jit.codeBlock()->addCallLinkInfo();
 
     JITCompiler::DataLabelPtr targetToCheck;
-    JITCompiler::Jump slowPath = m_jit.branchPtrWithPatch(MacroAssembler::NotEqual, calleeGPR, targetToCheck, MacroAssembler::TrustedImmPtr(0));
+    MacroAssembler::JumpList slowPath;
+    slowPath.append(m_jit.branchPtrWithPatch(MacroAssembler::NotEqual, calleeGPR, targetToCheck, MacroAssembler::TrustedImmPtr(0)));
 
 #if CPU(X32)
     slowPath.append(m_jit.branchTest64(MacroAssembler::NonZero, calleeGPR, GPRInfo::tagMaskRegister));
