diff -ur webkitgtk-2.12.3.orig/Source/JavaScriptCore/llint/LLIntData.cpp webkitgtk-2.12.3/Source/JavaScriptCore/llint/LLIntData.cpp
--- webkitgtk-2.12.3.orig/Source/JavaScriptCore/llint/LLIntData.cpp	2016-05-23 03:03:36.000000000 -0700
+++ webkitgtk-2.12.3/Source/JavaScriptCore/llint/LLIntData.cpp	2016-07-25 10:10:23.241867050 -0700
@@ -75,7 +75,11 @@
     // prepared to change LowLevelInterpreter.asm as well!!
 
 #if USE(JSVALUE64)
+#if CPU(X32)
+    const ptrdiff_t PtrSize = 4;
+#else
     const ptrdiff_t PtrSize = 8;
+#endif
     const ptrdiff_t CallFrameHeaderSlots = 5;
 #else // USE(JSVALUE64) // i.e. 32-bit version
     const ptrdiff_t PtrSize = 4;
