diff -ur webkitgtk-2.0.4.orig/Source/JavaScriptCore/offlineasm/x86.rb webkitgtk-2.0.4/Source/JavaScriptCore/offlineasm/x86.rb
--- webkitgtk-2.0.4.orig/Source/JavaScriptCore/offlineasm/x86.rb	2016-07-23 17:32:42.931305631 -0700
+++ webkitgtk-2.0.4/Source/JavaScriptCore/offlineasm/x86.rb	2016-07-23 17:51:52.196231466 -0700
@@ -278,6 +278,9 @@
     def x86CallOperand(kind)
         isX64 ? "*#{x86Operand(:quad)}" : "*#{x86Operand(:ptr)}"
     end
+    def x86PushOperand(kind)
+        isX64 ? "#{x86Operand(:quad)}" : "#{x86Operand(:ptr)}"
+    end
 end
 
 class FPRegisterID
