diff -ur webkitgtk-2.12.3.orig/Source/JavaScriptCore/jit/GPRInfo.h webkitgtk-2.12.3/Source/JavaScriptCore/jit/GPRInfo.h
--- webkitgtk-2.12.3.orig/Source/JavaScriptCore/jit/GPRInfo.h	2016-07-25 02:46:13.277586846 -0700
+++ webkitgtk-2.12.3/Source/JavaScriptCore/jit/GPRInfo.h	2016-07-25 02:45:00.359591551 -0700
@@ -894,8 +894,10 @@
 // and regT0 + regT1 (for 32-bit) serving as the accumulator register(s) for
 // passing results of one opcode to the next. Hence:
 COMPILE_ASSERT(GPRInfo::regT0 == GPRInfo::returnValueGPR, regT0_must_equal_returnValueGPR);
-#if USE(JSVALUE32_64)
+#if USE(JSVALUE32_64) && !CPU(X32)
 COMPILE_ASSERT(GPRInfo::regT1 == GPRInfo::returnValueGPR2, regT1_must_equal_returnValueGPR2);
+#elif USE(JSVALUE32_64) && CPU(X32)
+COMPILE_ASSERT(GPRInfo::regT2 == GPRInfo::returnValueGPR2, regT2_must_equal_returnValueGPR2);
 #endif
 
 #endif // ENABLE(JIT)
