diff -ur webkitgtk-2.0.4.orig/Source/WebKit2/UIProcess/Plugins/unix/PluginProcessProxyUnix.cpp webkitgtk-2.0.4/Source/WebKit2/UIProcess/Plugins/unix/PluginProcessProxyUnix.cpp
--- webkitgtk-2.0.4.orig/Source/WebKit2/UIProcess/Plugins/unix/PluginProcessProxyUnix.cpp	2016-07-22 14:18:20.410877560 -0700
+++ webkitgtk-2.0.4/Source/WebKit2/UIProcess/Plugins/unix/PluginProcessProxyUnix.cpp	2016-07-22 14:17:54.891877968 -0700
@@ -77,12 +77,14 @@
     // able to return the status.
     // As a consequence, we make sure that the disposition is set to
     // SIG_DFL before calling g_spawn_sync().
+#if defined(SIGCLD)
     struct sigaction action;
     sigaction(SIGCLD, 0, &action);
     if (action.sa_handler == SIG_IGN) {
         action.sa_handler = SIG_DFL;
         sigaction(SIGCLD, &action, 0);
     }
+#endif
 
     if (!g_spawn_sync(0, argv, 0, G_SPAWN_STDERR_TO_DEV_NULL, 0, 0, &stdOut, 0, &status, 0))
         return false;
