diff -ur webkitgtk-2.0.4.orig/Source/JavaScriptCore/offlineasm/cloop.rb webkitgtk-2.0.4/Source/JavaScriptCore/offlineasm/cloop.rb
--- webkitgtk-2.0.4.orig/Source/JavaScriptCore/offlineasm/cloop.rb	2013-07-19 00:03:48.000000000 -0700
+++ webkitgtk-2.0.4/Source/JavaScriptCore/offlineasm/cloop.rb	2016-07-22 09:19:37.168164475 -0700
@@ -398,7 +398,7 @@
 end
 
 def cloopEmitCompareDoubleWithNaNCheckAndBranch(operands, condition)
-    $asm.putc "if (std::isnan(#{operands[0].clValue(:double)}) || isnan(#{operands[1].clValue(:double)})"
+    $asm.putc "if (std::isnan(#{operands[0].clValue(:double)})"
     $asm.putc "    || (#{operands[0].clValue(:double)} #{condition} #{operands[1].clValue(:double)}))"
     $asm.putc "    goto #{operands[2].cLabel};"
 end
