diff -urN gdb-7.5.1.orig/gdb/debugreg.h gdb-7.5.1/gdb/debugreg.h
--- gdb-7.5.1.orig/gdb/debugreg.h	1969-12-31 16:00:00.000000000 -0800
+++ gdb-7.5.1/gdb/debugreg.h	2016-06-24 15:59:39.528416440 -0700
@@ -0,0 +1,49 @@
+#ifndef SYS_DEBUGREG_H
+#define SYS_DEBUGREG_H
+
+#include <stdint.h>
+
+#define DR_FIRSTADDR 0
+#define DR_LASTADDR 3
+
+#define DR_STATUS 6
+#define DR_CONTROL 7
+
+#define DR_TRAP0        (0x1)
+#define DR_TRAP1        (0x2)
+#define DR_TRAP2        (0x4)
+#define DR_TRAP3        (0x8)
+
+#define DR_STEP         (0x4000)
+#define DR_SWITCH       (0x8000)
+
+#define DR_CONTROL_SHIFT 16
+#define DR_CONTROL_SIZE  4
+
+#define DR_RW_EXECUTE   (0x0)
+#define DR_RW_WRITE     (0x1)
+#define DR_RW_READ      (0x3)
+
+#define DR_LEN_1 (0x0)
+#define DR_LEN_2 (0x4)
+#define DR_LEN_4 (0xC)
+#define DR_LEN_8 (0x8)
+
+#define DR_LOCAL_ENABLE_SHIFT  0
+#define DR_GLOBAL_ENABLE_SHIFT 1
+#define DR_ENABLE_SIZE         2
+
+#define DR_LOCAL_ENABLE_MASK  (0x55)
+#define DR_GLOBAL_ENABLE_MASK (0xAA)
+
+
+#if SIZE_MAX > 4294967295
+# define DR_CONTROL_RESERVED (0xFFFFFFFF0000FC00UL)
+#else
+# define DR_CONTROL_RESERVED (0x00FC00U)
+#endif
+#define DR_LOCAL_SLOWDOWN   (0x100)
+#define DR_GLOBAL_SLOWDOWN  (0x200)
+
+#endif
+
diff -urN gdb-7.5.1.orig/gdb/gdbserver/reg.h gdb-7.5.1/gdb/gdbserver/reg.h
--- gdb-7.5.1.orig/gdb/gdbserver/reg.h	1969-12-31 16:00:00.000000000 -0800
+++ gdb-7.5.1/gdb/gdbserver/reg.h	2016-06-24 16:00:37.646414579 -0700
@@ -0,0 +1,68 @@
+#ifndef REG_H_
+#define REG_H_
+
+#warning using REG H
+
+#ifdef __i386__
+
+#warning 386!
+
+#define EBX 0
+#define ECX 1
+#define EDX 2
+#define ESI 3
+#define EDI 4
+#define EBP 5
+#define EAX 6
+#define DS 7
+#define ES 8
+#define FS 9
+#define GS 10
+#define ORIG_EAX 11
+#define EIP 12
+#define CS  13
+#define EFL 14
+#define UESP 15
+#define SS   16
+#define FRAME_SIZE 17
+
+#else /* __i386__ */
+
+#warning not i386!
+
+#if defined(__ASSEMBLY__) || defined(__FRAME_OFFSETS)
+#define R15 0
+#define R14 8
+#define R13 16
+#define R12 24
+#define RBP 32
+#define RBX 40
+
+/* arguments: interrupts/non tracing syscalls only save upto here*/
+#define R11 48
+#define R10 56
+#define R9 64
+#define R8 72
+#define RAX 80
+#define RCX 88
+#define RDX 96
+#define RSI 104
+#define RDI 112
+#define ORIG_RAX 120       /* = ERROR */
+/* end of arguments */
+/* cpu exception frame or undefined in case of fast syscall. */
+#define RIP 128
+#define CS 136
+#define EFLAGS 144
+#define RSP 152
+#define SS 160
+#define ARGOFFSET R11
+#endif /* __ASSEMBLY__ */
+
+/* top of stack page */
+#define FRAME_SIZE 168
+
+#endif /* !__i386__ */
+
+#endif
+
diff -urN gdb-7.5.1.orig/gdb/threadpathhack.h gdb-7.5.1/gdb/threadpathhack.h
--- gdb-7.5.1.orig/gdb/threadpathhack.h	1969-12-31 16:00:00.000000000 -0800
+++ gdb-7.5.1/gdb/threadpathhack.h	2016-06-24 16:01:03.434413753 -0700
@@ -0,0 +1,12 @@
+#ifndef HACK_PATH_SO
+#define HACK_PATH_SO
+
+#ifndef LIBTHREAD_DB_SO
+#define LIBTHREAD_DB_SO "bug.so"
+#endif
+#ifndef LIBTHREAD_DB_SEARCH_PATH
+#define LIBTHREAD_DB_SEARCH_PATH "/lib"
+#endif
+
+#endif
+
diff -urN gdb-7.5.1.orig/gdb/stopcode.h gdb-7.5.1/gdb/stopcode.h
--- gdb-7.5.1.orig/gdb/stopcode.h	1969-12-31 16:00:00.000000000 -0800
+++ gdb-7.5.1/gdb/stopcode.h	2016-06-24 16:01:03.434413753 -0700
@@ -0,0 +1,4 @@
+#ifndef W_STOPCODE
+#define W_STOPCODE(sig) ((sig) << 8 | 0x7f)
+#endif
+
